<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>Applied Population Genetics</title>
  <meta content="text/html; charset=UTF-8" http-equiv="Content-Type">
  <meta name="description" content="A textbook for the use of R in spatial genetic data analysis.">
  <meta name="generator" content="bookdown 0.3 and GitBook 2.6.7">

  <meta property="og:title" content="Applied Population Genetics" />
  <meta property="og:type" content="book" />
  
  <meta property="og:image" content="media/TitlePage_small.png" />
  <meta property="og:description" content="A textbook for the use of R in spatial genetic data analysis." />
  <meta name="github-repo" content="dyerlab/AMG" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Applied Population Genetics" />
  
  <meta name="twitter:description" content="A textbook for the use of R in spatial genetic data analysis." />
  <meta name="twitter:image" content="media/TitlePage_small.png" />

<meta name="author" content="Rodney J. Dyer">



  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="hardy-weinberg-equilibrium.html">
<link rel="next" href="effective-population-size.html">

<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />







<script src="libs/htmlwidgets-0.8/htmlwidgets.js"></script>
<link href="libs/leaflet-0.7.7/leaflet.css" rel="stylesheet" />
<script src="libs/leaflet-0.7.7/leaflet.js"></script>
<link href="libs/leafletfix-1.0.0/leafletfix.css" rel="stylesheet" />
<link href="libs/leaflet-label-0.2.2/leaflet.label.css" rel="stylesheet" />
<script src="libs/leaflet-label-0.2.2/leaflet.label.js"></script>
<script src="libs/Proj4Leaflet-0.7.2/proj4-compressed.js"></script>
<script src="libs/Proj4Leaflet-0.7.2/proj4leaflet.js"></script>
<script src="libs/leaflet-binding-1.1.0/leaflet.js"></script>
<script src="libs/leaflet-providers-1.0.27/leaflet-providers.js"></script>
<script src="libs/leaflet-providers-plugin-1.1.0/leaflet-providers-plugin.js"></script>
<script src="libs/d3-4.5.0/d3.min.js"></script>
<script src="libs/sankey-1/sankey.js"></script>
<script src="libs/sankeyNetwork-binding-0.3/sankeyNetwork.js"></script>
<script src="libs/dendroNetwork-binding-0.3/dendroNetwork.js"></script>
<script src="libs/simpleNetwork-binding-0.3/simpleNetwork.js"></script>
<script src="libs/forceNetwork-binding-0.3/forceNetwork.js"></script>


<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; background-color: #f8f8f8; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
pre, code { background-color: #f8f8f8; }
code > span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code > span.dt { color: #204a87; } /* DataType */
code > span.dv { color: #0000cf; } /* DecVal */
code > span.bn { color: #0000cf; } /* BaseN */
code > span.fl { color: #0000cf; } /* Float */
code > span.ch { color: #4e9a06; } /* Char */
code > span.st { color: #4e9a06; } /* String */
code > span.co { color: #8f5902; font-style: italic; } /* Comment */
code > span.ot { color: #8f5902; } /* Other */
code > span.al { color: #ef2929; } /* Alert */
code > span.fu { color: #000000; } /* Function */
code > span.er { color: #a40000; font-weight: bold; } /* Error */
code > span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #000000; } /* Constant */
code > span.sc { color: #000000; } /* SpecialChar */
code > span.vs { color: #4e9a06; } /* VerbatimString */
code > span.ss { color: #4e9a06; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #000000; } /* Variable */
code > span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code > span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code > span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code > span.ex { } /* Extension */
code > span.at { color: #c4a000; } /* Attribute */
code > span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code > span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
</style>

<link rel="stylesheet" href="media/style.css" type="text/css" />
</head>

<body>


  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Applied Population Genetics</a><ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#logistics"><i class="fa fa-check"></i>Logistics</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#motivation"><i class="fa fa-check"></i>Motivation</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#typography"><i class="fa fa-check"></i>Typography</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#interactive-content"><i class="fa fa-check"></i>Interactive Content</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#dedication"><i class="fa fa-check"></i>Dedication</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="the-r-ecosystem.html"><a href="the-r-ecosystem.html"><i class="fa fa-check"></i><b>1</b> The R Ecosystem</a><ul>
<li class="chapter" data-level="1.1" data-path="the-r-ecosystem.html"><a href="the-r-ecosystem.html#getting-r-configured"><i class="fa fa-check"></i><b>1.1</b> Getting R Configured</a><ul>
<li class="chapter" data-level="1.1.1" data-path="the-r-ecosystem.html"><a href="the-r-ecosystem.html#packages"><i class="fa fa-check"></i><b>1.1.1</b> Packages</a></li>
<li class="chapter" data-level="1.1.2" data-path="the-r-ecosystem.html"><a href="the-r-ecosystem.html#cran"><i class="fa fa-check"></i><b>1.1.2</b> CRAN</a></li>
<li class="chapter" data-level="1.1.3" data-path="the-r-ecosystem.html"><a href="the-r-ecosystem.html#github"><i class="fa fa-check"></i><b>1.1.3</b> GitHub</a></li>
<li class="chapter" data-level="1.1.4" data-path="the-r-ecosystem.html"><a href="the-r-ecosystem.html#bioconductor"><i class="fa fa-check"></i><b>1.1.4</b> Bioconductor</a></li>
<li class="chapter" data-level="1.1.5" data-path="the-r-ecosystem.html"><a href="the-r-ecosystem.html#troublesome-packages"><i class="fa fa-check"></i><b>1.1.5</b> Troublesome Packages</a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="the-r-ecosystem.html"><a href="the-r-ecosystem.html#libraries-used-in-text"><i class="fa fa-check"></i><b>1.2</b> Libraries Used in Text</a></li>
<li class="chapter" data-level="1.3" data-path="the-r-ecosystem.html"><a href="the-r-ecosystem.html#the-rstudio-environment"><i class="fa fa-check"></i><b>1.3</b> The RStudio Environment</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="data-types.html"><a href="data-types.html"><i class="fa fa-check"></i><b>2</b> Data Types</a><ul>
<li class="chapter" data-level="2.1" data-path="data-types.html"><a href="data-types.html#numeric-data-types"><i class="fa fa-check"></i><b>2.1</b> Numeric Data Types</a><ul>
<li class="chapter" data-level="2.1.1" data-path="data-types.html"><a href="data-types.html#coercion-to-numeric"><i class="fa fa-check"></i><b>2.1.1</b> Coercion to Numeric</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="data-types.html"><a href="data-types.html#characters"><i class="fa fa-check"></i><b>2.2</b> Characters</a><ul>
<li class="chapter" data-level="2.2.1" data-path="data-types.html"><a href="data-types.html#concatenation-of-characters"><i class="fa fa-check"></i><b>2.2.1</b> Concatenation of Characters</a></li>
<li class="chapter" data-level="2.2.2" data-path="data-types.html"><a href="data-types.html#coercion-to-characters"><i class="fa fa-check"></i><b>2.2.2</b> Coercion to Characters</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="data-types.html"><a href="data-types.html#factors"><i class="fa fa-check"></i><b>2.3</b> Factors</a><ul>
<li class="chapter" data-level="2.3.1" data-path="data-types.html"><a href="data-types.html#missing-levels-in-factors"><i class="fa fa-check"></i><b>2.3.1</b> Missing Levels in Factors</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="data-types.html"><a href="data-types.html#logical-types"><i class="fa fa-check"></i><b>2.4</b> Logical Types</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="data-containers.html"><a href="data-containers.html"><i class="fa fa-check"></i><b>3</b> Data Containers</a><ul>
<li class="chapter" data-level="3.1" data-path="data-containers.html"><a href="data-containers.html#vectors"><i class="fa fa-check"></i><b>3.1</b> Vectors</a></li>
<li class="chapter" data-level="3.2" data-path="data-containers.html"><a href="data-containers.html#matrices"><i class="fa fa-check"></i><b>3.2</b> Matrices</a></li>
<li class="chapter" data-level="3.3" data-path="data-containers.html"><a href="data-containers.html#lists"><i class="fa fa-check"></i><b>3.3</b> Lists</a></li>
<li class="chapter" data-level="3.4" data-path="data-containers.html"><a href="data-containers.html#data-frames"><i class="fa fa-check"></i><b>3.4</b> Data Frames</a><ul>
<li class="chapter" data-level="3.4.1" data-path="data-containers.html"><a href="data-containers.html#indexing-data-frames"><i class="fa fa-check"></i><b>3.4.1</b> Indexing Data Frames</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="programming.html"><a href="programming.html"><i class="fa fa-check"></i><b>4</b> Programming</a><ul>
<li class="chapter" data-level="4.1" data-path="programming.html"><a href="programming.html#function-writing"><i class="fa fa-check"></i><b>4.1</b> Function Writing</a></li>
<li class="chapter" data-level="4.2" data-path="programming.html"><a href="programming.html#variable-scope"><i class="fa fa-check"></i><b>4.2</b> Variable Scope</a></li>
<li class="chapter" data-level="4.3" data-path="programming.html"><a href="programming.html#decision-making"><i class="fa fa-check"></i><b>4.3</b> Decision Making</a><ul>
<li class="chapter" data-level="4.3.1" data-path="programming.html"><a href="programming.html#the-if-pattern"><i class="fa fa-check"></i><b>4.3.1</b> The if Pattern</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="programming.html"><a href="programming.html#the-if-else-pattern"><i class="fa fa-check"></i><b>4.4</b> The <code>if-else</code> Pattern</a></li>
<li class="chapter" data-level="4.5" data-path="programming.html"><a href="programming.html#flow-control"><i class="fa fa-check"></i><b>4.5</b> Flow Control</a><ul>
<li class="chapter" data-level="4.5.1" data-path="programming.html"><a href="programming.html#the-for-loop"><i class="fa fa-check"></i><b>4.5.1</b> The for() Loop</a></li>
<li class="chapter" data-level="4.5.2" data-path="programming.html"><a href="programming.html#short-circuiting-the-loop"><i class="fa fa-check"></i><b>4.5.2</b> Short Circuiting the Loop</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="graphics.html"><a href="graphics.html"><i class="fa fa-check"></i><b>5</b> Graphics</a><ul>
<li class="chapter" data-level="5.1" data-path="graphics.html"><a href="graphics.html#univariate-data"><i class="fa fa-check"></i><b>5.1</b> Univariate Data</a></li>
<li class="chapter" data-level="5.2" data-path="graphics.html"><a href="graphics.html#plotting-parameters"><i class="fa fa-check"></i><b>5.2</b> Plotting Parameters</a></li>
<li class="chapter" data-level="5.3" data-path="graphics.html"><a href="graphics.html#density-plots"><i class="fa fa-check"></i><b>5.3</b> Density Plots</a></li>
<li class="chapter" data-level="5.4" data-path="graphics.html"><a href="graphics.html#overlaying-plots"><i class="fa fa-check"></i><b>5.4</b> Overlaying Plots</a></li>
<li class="chapter" data-level="5.5" data-path="graphics.html"><a href="graphics.html#boxplots"><i class="fa fa-check"></i><b>5.5</b> Boxplots</a></li>
<li class="chapter" data-level="5.6" data-path="graphics.html"><a href="graphics.html#scatter-plots"><i class="fa fa-check"></i><b>5.6</b> Scatter Plots</a></li>
<li class="chapter" data-level="5.7" data-path="graphics.html"><a href="graphics.html#multiple-plots"><i class="fa fa-check"></i><b>5.7</b> Multiple Plots</a></li>
<li class="chapter" data-level="5.8" data-path="graphics.html"><a href="graphics.html#color-palettes"><i class="fa fa-check"></i><b>5.8</b> Color Palettes</a></li>
<li class="chapter" data-level="5.9" data-path="graphics.html"><a href="graphics.html#rcolorbrewer"><i class="fa fa-check"></i><b>5.9</b> RColorBrewer</a></li>
<li class="chapter" data-level="5.10" data-path="graphics.html"><a href="graphics.html#saving-imagery"><i class="fa fa-check"></i><b>5.10</b> Saving Imagery</a></li>
<li class="chapter" data-level="5.11" data-path="graphics.html"><a href="graphics.html#interactive-graphics"><i class="fa fa-check"></i><b>5.11</b> Interactive Graphics</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="reproducable-research.html"><a href="reproducable-research.html"><i class="fa fa-check"></i><b>6</b> Reproducable Research</a><ul>
<li class="chapter" data-level="6.1" data-path="reproducable-research.html"><a href="reproducable-research.html#keeping-data-on-google-drive"><i class="fa fa-check"></i><b>6.1</b> Keeping Data on Google Drive</a></li>
<li class="chapter" data-level="6.2" data-path="reproducable-research.html"><a href="reproducable-research.html#integrating-into-slack"><i class="fa fa-check"></i><b>6.2</b> Integrating into Slack</a><ul>
<li class="chapter" data-level="6.2.1" data-path="reproducable-research.html"><a href="reproducable-research.html#fixing-problem-with-version-1.4.1"><i class="fa fa-check"></i><b>6.2.1</b> Fixing Problem with Version 1.4.1</a></li>
<li class="chapter" data-level="6.2.2" data-path="reproducable-research.html"><a href="reproducable-research.html#configuration"><i class="fa fa-check"></i><b>6.2.2</b> Configuration</a></li>
<li class="chapter" data-level="6.2.3" data-path="reproducable-research.html"><a href="reproducable-research.html#sending-text-and-graphical-ouput"><i class="fa fa-check"></i><b>6.2.3</b> Sending Text and Graphical Ouput</a></li>
<li class="chapter" data-level="6.2.4" data-path="reproducable-research.html"><a href="reproducable-research.html#markdown"><i class="fa fa-check"></i><b>6.2.4</b> Markdown</a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>I Single Population</b></span></li>
<li class="chapter" data-level="7" data-path="hardy-weinberg-equilibrium.html"><a href="hardy-weinberg-equilibrium.html"><i class="fa fa-check"></i><b>7</b> Hardy-Weinberg Equilibrium</a><ul>
<li class="chapter" data-level="7.1" data-path="hardy-weinberg-equilibrium.html"><a href="hardy-weinberg-equilibrium.html#genetic-assumptions"><i class="fa fa-check"></i><b>7.1</b> Genetic Assumptions</a><ul>
<li class="chapter" data-level="7.1.1" data-path="hardy-weinberg-equilibrium.html"><a href="hardy-weinberg-equilibrium.html#punnet-squares"><i class="fa fa-check"></i><b>7.1.1</b> Punnet Squares</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="hardy-weinberg-equilibrium.html"><a href="hardy-weinberg-equilibrium.html#evolutionary-assumptions"><i class="fa fa-check"></i><b>7.2</b> Evolutionary Assumptions</a></li>
<li class="chapter" data-level="7.3" data-path="hardy-weinberg-equilibrium.html"><a href="hardy-weinberg-equilibrium.html#the-mechanics"><i class="fa fa-check"></i><b>7.3</b> The Mechanics</a></li>
<li class="chapter" data-level="7.4" data-path="hardy-weinberg-equilibrium.html"><a href="hardy-weinberg-equilibrium.html#trans-generational-transition-probabilities"><i class="fa fa-check"></i><b>7.4</b> Trans-Generational Transition Probabilities</a></li>
<li class="chapter" data-level="7.5" data-path="hardy-weinberg-equilibrium.html"><a href="hardy-weinberg-equilibrium.html#consequences-of-hwe"><i class="fa fa-check"></i><b>7.5</b> Consequences of HWE</a></li>
<li class="chapter" data-level="7.6" data-path="hardy-weinberg-equilibrium.html"><a href="hardy-weinberg-equilibrium.html#zygotic-gene-flow"><i class="fa fa-check"></i><b>7.6</b> Zygotic Gene Flow</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="genetic-drift.html"><a href="genetic-drift.html"><i class="fa fa-check"></i><b>8</b> Genetic Drift</a><ul>
<li class="chapter" data-level="8.1" data-path="genetic-drift.html"><a href="genetic-drift.html#consequences-of-genetic-drift"><i class="fa fa-check"></i><b>8.1</b> Consequences of Genetic Drift</a><ul>
<li class="chapter" data-level="8.1.1" data-path="genetic-drift.html"><a href="genetic-drift.html#time-to-fixation"><i class="fa fa-check"></i><b>8.1.1</b> Time to Fixation</a></li>
<li class="chapter" data-level="8.1.2" data-path="genetic-drift.html"><a href="genetic-drift.html#time-to-allele-loss"><i class="fa fa-check"></i><b>8.1.2</b> Time to Allele Loss</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="genetic-drift.html"><a href="genetic-drift.html#mutation-drift"><i class="fa fa-check"></i><b>8.2</b> Mutation &amp; Drift</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="effective-population-size.html"><a href="effective-population-size.html"><i class="fa fa-check"></i><b>9</b> Effective Population Size</a><ul>
<li class="chapter" data-level="9.1" data-path="effective-population-size.html"><a href="effective-population-size.html#demographic-effective-population-sizes"><i class="fa fa-check"></i><b>9.1</b> Demographic Effective Population Sizes</a></li>
<li class="chapter" data-level="9.2" data-path="effective-population-size.html"><a href="effective-population-size.html#genetic-effective-population-size"><i class="fa fa-check"></i><b>9.2</b> Genetic Effective Population Size</a><ul>
<li class="chapter" data-level="9.2.1" data-path="effective-population-size.html"><a href="effective-population-size.html#inbreeding-effective-population-size"><i class="fa fa-check"></i><b>9.2.1</b> Inbreeding Effective Population Size</a></li>
<li class="chapter" data-level="9.2.2" data-path="effective-population-size.html"><a href="effective-population-size.html#variance-effective-population-size"><i class="fa fa-check"></i><b>9.2.2</b> Variance Effective Population Size</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="effective-population-size.html"><a href="effective-population-size.html#fluctuating-population-size"><i class="fa fa-check"></i><b>9.3</b> Fluctuating Population Size</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="mutation.html"><a href="mutation.html"><i class="fa fa-check"></i><b>10</b> Mutation</a><ul>
<li class="chapter" data-level="10.1" data-path="mutation.html"><a href="mutation.html#mutation-models"><i class="fa fa-check"></i><b>10.1</b> Mutation Models</a><ul>
<li class="chapter" data-level="10.1.1" data-path="mutation.html"><a href="mutation.html#fixed-allele-model"><i class="fa fa-check"></i><b>10.1.1</b> Fixed Allele Model</a></li>
<li class="chapter" data-level="10.1.2" data-path="mutation.html"><a href="mutation.html#infinite-allele-model"><i class="fa fa-check"></i><b>10.1.2</b> Infinite Allele Model</a></li>
<li class="chapter" data-level="10.1.3" data-path="mutation.html"><a href="mutation.html#stepwise-mutation-model"><i class="fa fa-check"></i><b>10.1.3</b> Stepwise Mutation Model</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="mutation.html"><a href="mutation.html#mutation-and-inbreeding"><i class="fa fa-check"></i><b>10.2</b> Mutation and Inbreeding</a></li>
<li class="chapter" data-level="10.3" data-path="mutation.html"><a href="mutation.html#estimating-mutation"><i class="fa fa-check"></i><b>10.3</b> Estimating Mutation</a><ul>
<li class="chapter" data-level="10.3.1" data-path="mutation.html"><a href="mutation.html#phenotypes"><i class="fa fa-check"></i><b>10.3.1</b> Phenotypes</a></li>
<li class="chapter" data-level="10.3.2" data-path="mutation.html"><a href="mutation.html#genotypes"><i class="fa fa-check"></i><b>10.3.2</b> Genotypes</a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>II Nonrandom Rating</b></span></li>
<li class="chapter" data-level="11" data-path="selfing.html"><a href="selfing.html"><i class="fa fa-check"></i><b>11</b> Selfing</a><ul>
<li class="chapter" data-level="11.1" data-path="selfing.html"><a href="selfing.html#changes-in-f"><i class="fa fa-check"></i><b>11.1</b> Changes in F</a><ul>
<li class="chapter" data-level="11.1.1" data-path="selfing.html"><a href="selfing.html#theoretical-expectations"><i class="fa fa-check"></i><b>11.1.1</b> Theoretical Expectations</a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="selfing.html"><a href="selfing.html#simulation-example"><i class="fa fa-check"></i><b>11.2</b> Simulation Example</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="mixed-mating-systems.html"><a href="mixed-mating-systems.html"><i class="fa fa-check"></i><b>12</b> Mixed Mating Systems</a><ul>
<li class="chapter" data-level="12.1" data-path="mixed-mating-systems.html"><a href="mixed-mating-systems.html#mixed-mating-genotype-frequency-equilibriums"><i class="fa fa-check"></i><b>12.1</b> Mixed Mating Genotype Frequency Equilibriums</a></li>
<li class="chapter" data-level="12.2" data-path="mixed-mating-systems.html"><a href="mixed-mating-systems.html#estimating-selfing-rates"><i class="fa fa-check"></i><b>12.2</b> Estimating Selfing Rates</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="wahlund-effects.html"><a href="wahlund-effects.html"><i class="fa fa-check"></i><b>13</b> Wahlund Effects</a></li>
<li class="part"><span><b>III Parent Offspring Data</b></span></li>
<li class="chapter" data-level="14" data-path="pedigrees.html"><a href="pedigrees.html"><i class="fa fa-check"></i><b>14</b> Pedigrees</a><ul>
<li class="chapter" data-level="14.1" data-path="pedigrees.html"><a href="pedigrees.html#expectations-for-f-in-a-pedigree."><i class="fa fa-check"></i><b>14.1</b> Expectations for F in a pedigree.</a></li>
<li class="chapter" data-level="14.2" data-path="pedigrees.html"><a href="pedigrees.html#drawing-pedigrees"><i class="fa fa-check"></i><b>14.2</b> Drawing Pedigrees</a></li>
<li class="chapter" data-level="14.3" data-path="pedigrees.html"><a href="pedigrees.html#skills"><i class="fa fa-check"></i><b>14.3</b> Skills</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="parentage-analyses.html"><a href="parentage-analyses.html"><i class="fa fa-check"></i><b>15</b> Parentage Analyses</a><ul>
<li class="chapter" data-level="15.1" data-path="parentage-analyses.html"><a href="parentage-analyses.html#paternity-vs.maternity"><i class="fa fa-check"></i><b>15.1</b> Paternity vs. Maternity</a></li>
<li class="chapter" data-level="15.2" data-path="parentage-analyses.html"><a href="parentage-analyses.html#probability-of-exclusion"><i class="fa fa-check"></i><b>15.2</b> Probability of Exclusion</a></li>
<li class="chapter" data-level="15.3" data-path="parentage-analyses.html"><a href="parentage-analyses.html#paternity-exclusion"><i class="fa fa-check"></i><b>15.3</b> Paternity Exclusion</a></li>
<li class="chapter" data-level="15.4" data-path="parentage-analyses.html"><a href="parentage-analyses.html#fractional-paternity"><i class="fa fa-check"></i><b>15.4</b> Fractional Paternity</a></li>
<li class="chapter" data-level="15.5" data-path="parentage-analyses.html"><a href="parentage-analyses.html#dispersal-kernels"><i class="fa fa-check"></i><b>15.5</b> Dispersal Kernels</a></li>
</ul></li>
<li class="part"><span><b>IV Subdivided Populations</b></span></li>
<li class="chapter" data-level="16" data-path="mapping-populations.html"><a href="mapping-populations.html"><i class="fa fa-check"></i><b>16</b> Mapping Populations</a><ul>
<li class="chapter" data-level="16.1" data-path="mapping-populations.html"><a href="mapping-populations.html#procedure"><i class="fa fa-check"></i><b>16.1</b> Procedure</a></li>
<li class="chapter" data-level="16.2" data-path="mapping-populations.html"><a href="mapping-populations.html#plotting-population-locations"><i class="fa fa-check"></i><b>16.2</b> Plotting Population Locations</a></li>
<li class="chapter" data-level="16.3" data-path="mapping-populations.html"><a href="mapping-populations.html#population-allele-frequencies"><i class="fa fa-check"></i><b>16.3</b> Population Allele Frequencies</a></li>
<li class="chapter" data-level="16.4" data-path="mapping-populations.html"><a href="mapping-populations.html#dynamic-maps"><i class="fa fa-check"></i><b>16.4</b> Dynamic maps</a><ul>
<li class="chapter" data-level="16.4.1" data-path="mapping-populations.html"><a href="mapping-populations.html#sankey-networks"><i class="fa fa-check"></i><b>16.4.1</b> Sankey Networks</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="17" data-path="population-models.html"><a href="population-models.html"><i class="fa fa-check"></i><b>17</b> Population Models</a><ul>
<li class="chapter" data-level="17.1" data-path="population-models.html"><a href="population-models.html#models-of-partition"><i class="fa fa-check"></i><b>17.1</b> Models of Partition</a></li>
<li class="chapter" data-level="17.2" data-path="population-models.html"><a href="population-models.html#island-mainland-model"><i class="fa fa-check"></i><b>17.2</b> Island Mainland Model</a></li>
<li class="chapter" data-level="17.3" data-path="population-models.html"><a href="population-models.html#the-island-model"><i class="fa fa-check"></i><b>17.3</b> The Island Model</a></li>
<li class="chapter" data-level="17.4" data-path="population-models.html"><a href="population-models.html#stepping-stone-models"><i class="fa fa-check"></i><b>17.4</b> Stepping Stone Models</a></li>
<li class="chapter" data-level="17.5" data-path="population-models.html"><a href="population-models.html#the-general-model"><i class="fa fa-check"></i><b>17.5</b> The General Model</a></li>
<li class="chapter" data-level="17.6" data-path="population-models.html"><a href="population-models.html#general-strategies"><i class="fa fa-check"></i><b>17.6</b> General Strategies</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="genetic-diversity.html"><a href="genetic-diversity.html"><i class="fa fa-check"></i><b>18</b> Genetic Diversity</a><ul>
<li class="chapter" data-level="18.1" data-path="genetic-diversity.html"><a href="genetic-diversity.html#allelic-diversity"><i class="fa fa-check"></i><b>18.1</b> Allelic Diversity</a></li>
<li class="chapter" data-level="18.2" data-path="genetic-diversity.html"><a href="genetic-diversity.html#genotypic-diversity"><i class="fa fa-check"></i><b>18.2</b> Genotypic Diversity</a></li>
<li class="chapter" data-level="18.3" data-path="genetic-diversity.html"><a href="genetic-diversity.html#samples-from-several-locales"><i class="fa fa-check"></i><b>18.3</b> Samples from Several Locales</a></li>
<li class="chapter" data-level="18.4" data-path="genetic-diversity.html"><a href="genetic-diversity.html#multilocus-diversity"><i class="fa fa-check"></i><b>18.4</b> Multilocus Diversity</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="rarefaction.html"><a href="rarefaction.html"><i class="fa fa-check"></i><b>19</b> Rarefaction</a><ul>
<li class="chapter" data-level="19.1" data-path="rarefaction.html"><a href="rarefaction.html#mapping-diversity"><i class="fa fa-check"></i><b>19.1</b> Mapping Diversity</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="genetic-distances.html"><a href="genetic-distances.html"><i class="fa fa-check"></i><b>20</b> Genetic Distances</a><ul>
<li class="chapter" data-level="20.1" data-path="genetic-distances.html"><a href="genetic-distances.html#individual-genetic-distance"><i class="fa fa-check"></i><b>20.1</b> Individual Genetic Distance</a><ul>
<li class="chapter" data-level="20.1.1" data-path="genetic-distances.html"><a href="genetic-distances.html#euclidean-distance"><i class="fa fa-check"></i><b>20.1.1</b> Euclidean Distance</a></li>
<li class="chapter" data-level="20.1.2" data-path="genetic-distances.html"><a href="genetic-distances.html#amova-distance"><i class="fa fa-check"></i><b>20.1.2</b> AMOVA Distance</a></li>
<li class="chapter" data-level="20.1.3" data-path="genetic-distances.html"><a href="genetic-distances.html#bray-curtis-individual-distance"><i class="fa fa-check"></i><b>20.1.3</b> Bray-Curtis Individual Distance</a></li>
<li class="chapter" data-level="20.1.4" data-path="genetic-distances.html"><a href="genetic-distances.html#ladder-distance"><i class="fa fa-check"></i><b>20.1.4</b> Ladder Distance</a></li>
</ul></li>
<li class="chapter" data-level="20.2" data-path="genetic-distances.html"><a href="genetic-distances.html#population-level-genetic-distances"><i class="fa fa-check"></i><b>20.2</b> Population-Level Genetic Distances</a><ul>
<li class="chapter" data-level="20.2.1" data-path="genetic-distances.html"><a href="genetic-distances.html#czekanowski-manhattan-distance"><i class="fa fa-check"></i><b>20.2.1</b> Czekanowski (Manhattan) Distance</a></li>
<li class="chapter" data-level="20.2.2" data-path="genetic-distances.html"><a href="genetic-distances.html#rogers-euclidean-distance"><i class="fa fa-check"></i><b>20.2.2</b> Rogers (Euclidean) Distance</a></li>
<li class="chapter" data-level="20.2.3" data-path="genetic-distances.html"><a href="genetic-distances.html#neis-genetic-distance"><i class="fa fa-check"></i><b>20.2.3</b> Nei’s Genetic Distance</a></li>
<li class="chapter" data-level="20.2.4" data-path="genetic-distances.html"><a href="genetic-distances.html#conditional-genetic-distance"><i class="fa fa-check"></i><b>20.2.4</b> Conditional Genetic Distance</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="21" data-path="genetic-structure.html"><a href="genetic-structure.html"><i class="fa fa-check"></i><b>21</b> Genetic Structure</a><ul>
<li class="chapter" data-level="21.1" data-path="genetic-structure.html"><a href="genetic-structure.html#f-statistics-and-thier-ilk"><i class="fa fa-check"></i><b>21.1</b> F-Statistics and Thier Ilk</a><ul>
<li class="chapter" data-level="21.1.1" data-path="genetic-structure.html"><a href="genetic-structure.html#issues-with-fixation-indices"><i class="fa fa-check"></i><b>21.1.1</b> Issues with Fixation Indices</a></li>
<li class="chapter" data-level="21.1.2" data-path="genetic-structure.html"><a href="genetic-structure.html#additional-x_st-like-parameters"><i class="fa fa-check"></i><b>21.1.2</b> Additional <span class="math inline">\(X_{ST}\)</span>-like Parameters</a></li>
</ul></li>
<li class="chapter" data-level="21.2" data-path="genetic-structure.html"><a href="genetic-structure.html#statistical-structure"><i class="fa fa-check"></i><b>21.2</b> Statistical Structure</a></li>
<li class="chapter" data-level="21.3" data-path="genetic-structure.html"><a href="genetic-structure.html#final-thoughts-on-structure"><i class="fa fa-check"></i><b>21.3</b> Final Thoughts on Structure</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="population-assignment.html"><a href="population-assignment.html"><i class="fa fa-check"></i><b>22</b> Population Assignment</a></li>
<li class="chapter" data-level="23" data-path="admixture.html"><a href="admixture.html"><i class="fa fa-check"></i><b>23</b> Admixture</a></li>
<li class="part"><span><b>V Statistical Models</b></span></li>
<li class="chapter" data-level="24" data-path="ordination-of-genetic-data.html"><a href="ordination-of-genetic-data.html"><i class="fa fa-check"></i><b>24</b> Ordination of Genetic Data</a><ul>
<li class="chapter" data-level="24.1" data-path="ordination-of-genetic-data.html"><a href="ordination-of-genetic-data.html#hierarchical-clustering"><i class="fa fa-check"></i><b>24.1</b> Hierarchical Clustering</a></li>
<li class="chapter" data-level="24.2" data-path="ordination-of-genetic-data.html"><a href="ordination-of-genetic-data.html#interactive-tree-display"><i class="fa fa-check"></i><b>24.2</b> Interactive Tree Display</a></li>
</ul></li>
<li class="chapter" data-level="25" data-path="distance-matrices.html"><a href="distance-matrices.html"><i class="fa fa-check"></i><b>25</b> Distance Matrices</a><ul>
<li class="chapter" data-level="25.1" data-path="distance-matrices.html"><a href="distance-matrices.html#visual-comparisons"><i class="fa fa-check"></i><b>25.1</b> Visual Comparisons</a></li>
<li class="chapter" data-level="25.2" data-path="distance-matrices.html"><a href="distance-matrices.html#the-mantel-test"><i class="fa fa-check"></i><b>25.2</b> The Mantel Test</a></li>
</ul></li>
<li class="chapter" data-level="26" data-path="linear-models.html"><a href="linear-models.html"><i class="fa fa-check"></i><b>26</b> Linear Models</a><ul>
<li class="chapter" data-level="26.1" data-path="linear-models.html"><a href="linear-models.html#dependent-independent-relationships"><i class="fa fa-check"></i><b>26.1</b> Dependent &amp; Independent Relationships</a></li>
</ul></li>
<li class="chapter" data-level="27" data-path="distance-networks.html"><a href="distance-networks.html"><i class="fa fa-check"></i><b>27</b> Distance Networks</a><ul>
<li class="chapter" data-level="27.1" data-path="distance-networks.html"><a href="distance-networks.html#graph-structure"><i class="fa fa-check"></i><b>27.1</b> Graph Structure</a><ul>
<li class="chapter" data-level="27.1.1" data-path="distance-networks.html"><a href="distance-networks.html#node-edge-attributes"><i class="fa fa-check"></i><b>27.1.1</b> Node &amp; Edge Attributes</a></li>
</ul></li>
<li class="chapter" data-level="27.2" data-path="distance-networks.html"><a href="distance-networks.html#plotting-a-graphs"><i class="fa fa-check"></i><b>27.2</b> Plotting a Graphs</a></li>
<li class="chapter" data-level="27.3" data-path="distance-networks.html"><a href="distance-networks.html#mapping-networks"><i class="fa fa-check"></i><b>27.3</b> Mapping Networks</a></li>
<li class="chapter" data-level="27.4" data-path="distance-networks.html"><a href="distance-networks.html#genetic-distance-graphs"><i class="fa fa-check"></i><b>27.4</b> Genetic Distance Graphs</a></li>
</ul></li>
<li class="chapter" data-level="28" data-path="population-graphs.html"><a href="population-graphs.html"><i class="fa fa-check"></i><b>28</b> Population Graphs</a><ul>
<li class="chapter" data-level="28.1" data-path="population-graphs.html"><a href="population-graphs.html#adding-data-to-a-graph"><i class="fa fa-check"></i><b>28.1</b> Adding data to a graph</a><ul>
<li class="chapter" data-level="28.1.1" data-path="population-graphs.html"><a href="population-graphs.html#plotting-a-graph-using-ggplot2-routines"><i class="fa fa-check"></i><b>28.1.1</b> Plotting a graph using <strong>ggplot2</strong> routines</a></li>
<li class="chapter" data-level="28.1.2" data-path="population-graphs.html"><a href="population-graphs.html#reading-existing-popgraph-files"><i class="fa fa-check"></i><b>28.1.2</b> Reading Existing <strong>popgraph</strong> Files</a></li>
<li class="chapter" data-level="28.1.3" data-path="population-graphs.html"><a href="population-graphs.html#saving-population-graph-objects"><i class="fa fa-check"></i><b>28.1.3</b> Saving Population Graph Objects</a></li>
</ul></li>
<li class="chapter" data-level="28.2" data-path="population-graphs.html"><a href="population-graphs.html#interactive-networks"><i class="fa fa-check"></i><b>28.2</b> Interactive Networks</a></li>
<li class="chapter" data-level="28.3" data-path="population-graphs.html"><a href="population-graphs.html#spatial-population-graphs"><i class="fa fa-check"></i><b>28.3</b> Spatial Population Graphs</a><ul>
<li class="chapter" data-level="28.3.1" data-path="population-graphs.html"><a href="population-graphs.html#integrating-google-and-ggplot2-for-plotting"><i class="fa fa-check"></i><b>28.3.1</b> Integrating Google and <strong>ggplot2</strong> for Plotting</a></li>
<li class="chapter" data-level="28.3.2" data-path="population-graphs.html"><a href="population-graphs.html#integrating-raster-maps"><i class="fa fa-check"></i><b>28.3.2</b> Integrating Raster Maps</a></li>
</ul></li>
<li class="chapter" data-level="28.4" data-path="population-graphs.html"><a href="population-graphs.html#extracting-spatial-data-using-population-graphs"><i class="fa fa-check"></i><b>28.4</b> Extracting Spatial Data Using Population Graphs</a><ul>
<li class="chapter" data-level="28.4.1" data-path="population-graphs.html"><a href="population-graphs.html#node-specific-data"><i class="fa fa-check"></i><b>28.4.1</b> Node Specific Data</a></li>
<li class="chapter" data-level="28.4.2" data-path="population-graphs.html"><a href="population-graphs.html#extracting-data-along-popgraph-edges"><i class="fa fa-check"></i><b>28.4.2</b> Extracting Data Along Popgraph Edges</a></li>
</ul></li>
<li class="chapter" data-level="28.5" data-path="population-graphs.html"><a href="population-graphs.html#extracting-graph-theoretic-parameters"><i class="fa fa-check"></i><b>28.5</b> Extracting Graph-Theoretic Parameters</a><ul>
<li class="chapter" data-level="28.5.1" data-path="population-graphs.html"><a href="population-graphs.html#matrix-representations-of-population-graph-objects"><i class="fa fa-check"></i><b>28.5.1</b> Matrix Representations of Population Graph Objects</a></li>
<li class="chapter" data-level="28.5.2" data-path="population-graphs.html"><a href="population-graphs.html#node-specific-parameters"><i class="fa fa-check"></i><b>28.5.2</b> Node Specific Parameters</a></li>
<li class="chapter" data-level="28.5.3" data-path="population-graphs.html"><a href="population-graphs.html#edge-specific-parameters"><i class="fa fa-check"></i><b>28.5.3</b> Edge Specific Parameters</a></li>
</ul></li>
<li class="chapter" data-level="28.6" data-path="population-graphs.html"><a href="population-graphs.html#testing-for-topological-congruence"><i class="fa fa-check"></i><b>28.6</b> Testing for Topological Congruence</a></li>
</ul></li>
<li class="chapter" data-level="29" data-path="interactive-visualization.html"><a href="interactive-visualization.html"><i class="fa fa-check"></i><b>29</b> Interactive Visualization</a><ul>
<li class="chapter" data-level="29.1" data-path="interactive-visualization.html"><a href="interactive-visualization.html#procedure-1"><i class="fa fa-check"></i><b>29.1</b> Procedure</a><ul>
<li class="chapter" data-level="29.1.1" data-path="interactive-visualization.html"><a href="interactive-visualization.html#interactive-networks-1"><i class="fa fa-check"></i><b>29.1.1</b> Interactive Networks</a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>VI Spatial Data</b></span></li>
<li class="chapter" data-level="30" data-path="vector-data.html"><a href="vector-data.html"><i class="fa fa-check"></i><b>30</b> Vector Data</a><ul>
<li class="chapter" data-level="30.1" data-path="vector-data.html"><a href="vector-data.html#points"><i class="fa fa-check"></i><b>30.1</b> Points</a></li>
<li class="chapter" data-level="30.2" data-path="vector-data.html"><a href="vector-data.html#lines"><i class="fa fa-check"></i><b>30.2</b> Lines</a></li>
<li class="chapter" data-level="30.3" data-path="vector-data.html"><a href="vector-data.html#polygons"><i class="fa fa-check"></i><b>30.3</b> Polygons</a></li>
<li class="chapter" data-level="30.4" data-path="vector-data.html"><a href="vector-data.html#saving-vector-objects"><i class="fa fa-check"></i><b>30.4</b> Saving Vector Objects</a></li>
</ul></li>
<li class="chapter" data-level="31" data-path="raster-data.html"><a href="raster-data.html"><i class="fa fa-check"></i><b>31</b> Raster Data</a><ul>
<li class="chapter" data-level="31.1" data-path="raster-data.html"><a href="raster-data.html#cropping-rasters"><i class="fa fa-check"></i><b>31.1</b> Cropping Rasters</a><ul>
<li class="chapter" data-level="31.1.1" data-path="raster-data.html"><a href="raster-data.html#cropping-rasters-via-polygons"><i class="fa fa-check"></i><b>31.1.1</b> Cropping Rasters Via Polygons</a></li>
<li class="chapter" data-level="31.1.2" data-path="raster-data.html"><a href="raster-data.html#cropping-rasters-via-convex-hull"><i class="fa fa-check"></i><b>31.1.2</b> Cropping Rasters Via Convex Hull</a></li>
</ul></li>
<li class="chapter" data-level="31.2" data-path="raster-data.html"><a href="raster-data.html#modifying-rasters"><i class="fa fa-check"></i><b>31.2</b> Modifying Rasters</a></li>
<li class="chapter" data-level="31.3" data-path="raster-data.html"><a href="raster-data.html#extracting-point-data-from-rasters"><i class="fa fa-check"></i><b>31.3</b> Extracting Point Data From Rasters</a><ul>
<li class="chapter" data-level="31.3.1" data-path="raster-data.html"><a href="raster-data.html#stacks-of-rasters"><i class="fa fa-check"></i><b>31.3.1</b> Stacks of Rasters</a></li>
</ul></li>
<li class="chapter" data-level="31.4" data-path="raster-data.html"><a href="raster-data.html#rasters-ggplot"><i class="fa fa-check"></i><b>31.4</b> Rasters &amp; ggplot</a></li>
<li class="chapter" data-level="31.5" data-path="raster-data.html"><a href="raster-data.html#d-visualization"><i class="fa fa-check"></i><b>31.5</b> 3D Visualization</a></li>
<li class="chapter" data-level="31.6" data-path="raster-data.html"><a href="raster-data.html#saving-exporting-rasters"><i class="fa fa-check"></i><b>31.6</b> Saving &amp; Exporting Rasters</a></li>
</ul></li>
<li class="chapter" data-level="32" data-path="map-projections.html"><a href="map-projections.html"><i class="fa fa-check"></i><b>32</b> Map Projections</a><ul>
<li class="chapter" data-level="32.1" data-path="map-projections.html"><a href="map-projections.html#projections"><i class="fa fa-check"></i><b>32.1</b> Projections</a><ul>
<li class="chapter" data-level="32.1.1" data-path="map-projections.html"><a href="map-projections.html#equatorial-projections"><i class="fa fa-check"></i><b>32.1.1</b> Equatorial Projections</a></li>
<li class="chapter" data-level="32.1.2" data-path="map-projections.html"><a href="map-projections.html#azimuth-projections"><i class="fa fa-check"></i><b>32.1.2</b> Azimuth Projections</a></li>
<li class="chapter" data-level="32.1.3" data-path="map-projections.html"><a href="map-projections.html#polar-conic-projections"><i class="fa fa-check"></i><b>32.1.3</b> Polar Conic Projections</a></li>
<li class="chapter" data-level="32.1.4" data-path="map-projections.html"><a href="map-projections.html#miscellaneous-projections"><i class="fa fa-check"></i><b>32.1.4</b> Miscellaneous Projections</a></li>
<li class="chapter" data-level="32.1.5" data-path="map-projections.html"><a href="map-projections.html#reprojecting-rasters"><i class="fa fa-check"></i><b>32.1.5</b> Reprojecting Rasters</a></li>
<li class="chapter" data-level="32.1.6" data-path="map-projections.html"><a href="map-projections.html#projecting-in-ggplot"><i class="fa fa-check"></i><b>32.1.6</b> Projecting In GGPlot</a></li>
</ul></li>
<li class="chapter" data-level="32.2" data-path="map-projections.html"><a href="map-projections.html#coordinate-systems"><i class="fa fa-check"></i><b>32.2</b> Coordinate Systems</a><ul>
<li class="chapter" data-level="32.2.1" data-path="map-projections.html"><a href="map-projections.html#changing-datum"><i class="fa fa-check"></i><b>32.2.1</b> Changing Datum</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="33" data-path="ecological-distance.html"><a href="ecological-distance.html"><i class="fa fa-check"></i><b>33</b> Ecological Distance</a><ul>
<li class="chapter" data-level="33.1" data-path="ecological-distance.html"><a href="ecological-distance.html#the-concept-of-resistance"><i class="fa fa-check"></i><b>33.1</b> The Concept of Resistance</a><ul>
<li class="chapter" data-level="33.1.1" data-path="ecological-distance.html"><a href="ecological-distance.html#absolute-resistance"><i class="fa fa-check"></i><b>33.1.1</b> Absolute Resistance</a></li>
</ul></li>
<li class="chapter" data-level="33.2" data-path="ecological-distance.html"><a href="ecological-distance.html#relative-resistance"><i class="fa fa-check"></i><b>33.2</b> Relative Resistance</a></li>
<li class="chapter" data-level="33.3" data-path="ecological-distance.html"><a href="ecological-distance.html#estimating-separation"><i class="fa fa-check"></i><b>33.3</b> Estimating Separation</a><ul>
<li class="chapter" data-level="33.3.1" data-path="ecological-distance.html"><a href="ecological-distance.html#least-cost-distance"><i class="fa fa-check"></i><b>33.3.1</b> Least Cost Distance</a></li>
<li class="chapter" data-level="33.3.2" data-path="ecological-distance.html"><a href="ecological-distance.html#all-paths-circuit-distance"><i class="fa fa-check"></i><b>33.3.2</b> All Paths (Circuit) Distance</a></li>
<li class="chapter" data-level="33.3.3" data-path="ecological-distance.html"><a href="ecological-distance.html#path-corridor-constraints"><i class="fa fa-check"></i><b>33.3.3</b> Path &amp; Corridor Constraints</a></li>
</ul></li>
<li class="chapter" data-level="33.4" data-path="ecological-distance.html"><a href="ecological-distance.html#estimating-correlations-in-distance"><i class="fa fa-check"></i><b>33.4</b> Estimating Correlations in Distance</a><ul>
<li class="chapter" data-level="33.4.1" data-path="ecological-distance.html"><a href="ecological-distance.html#pairwise-least-cost-path"><i class="fa fa-check"></i><b>33.4.1</b> Pairwise Least Cost Path</a></li>
<li class="chapter" data-level="33.4.2" data-path="ecological-distance.html"><a href="ecological-distance.html#pairwise-all-pairs"><i class="fa fa-check"></i><b>33.4.2</b> Pairwise All Pairs</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="34" data-path="niche-modeling.html"><a href="niche-modeling.html"><i class="fa fa-check"></i><b>34</b> Niche Modeling</a></li>
<li class="chapter" data-level="35" data-path="eigenvector-maps.html"><a href="eigenvector-maps.html"><i class="fa fa-check"></i><b>35</b> Eigenvector Maps</a><ul>
<li class="chapter" data-level="35.1" data-path="eigenvector-maps.html"><a href="eigenvector-maps.html#eigenvectors-eigenvalues"><i class="fa fa-check"></i><b>35.1</b> Eigenvectors &amp; Eigenvalues</a></li>
<li class="chapter" data-level="35.2" data-path="eigenvector-maps.html"><a href="eigenvector-maps.html#principal-coordinates-of-neighborhood-matrices."><i class="fa fa-check"></i><b>35.2</b> Principal Coordinates of Neighborhood Matrices.</a></li>
</ul></li>
<li class="part"><span><b>VII Selection</b></span></li>
<li class="chapter" data-level="36" data-path="classical-selection-models.html"><a href="classical-selection-models.html"><i class="fa fa-check"></i><b>36</b> Classical Selection Models</a><ul>
<li class="chapter" data-level="36.1" data-path="classical-selection-models.html"><a href="classical-selection-models.html#the-strength-of-selection"><i class="fa fa-check"></i><b>36.1</b> The Strength of Selection</a></li>
</ul></li>
<li class="part"><span><b>VIII Backmatter</b></span></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="chapter" data-level="37" data-path="included-data-sets.html"><a href="included-data-sets.html"><i class="fa fa-check"></i><b>37</b> Included Data Sets</a><ul>
<li class="chapter" data-level="37.1" data-path="included-data-sets.html"><a href="included-data-sets.html#the-sonoran-desert-bark-beetle-araptus-attenuata"><i class="fa fa-check"></i><b>37.1</b> The Sonoran desert bark beetle, <em>Araptus attenuata</em></a><ul>
<li class="chapter" data-level="37.1.1" data-path="included-data-sets.html"><a href="included-data-sets.html#nuclear-markers"><i class="fa fa-check"></i><b>37.1.1</b> Nuclear Markers</a></li>
<li class="chapter" data-level="37.1.2" data-path="included-data-sets.html"><a href="included-data-sets.html#methylation-markers"><i class="fa fa-check"></i><b>37.1.2</b> Methylation Markers</a></li>
</ul></li>
<li class="chapter" data-level="37.2" data-path="included-data-sets.html"><a href="included-data-sets.html#the-flowering-dogwood-cornus-florida"><i class="fa fa-check"></i><b>37.2</b> The Flowering Dogwood <em>Cornus florida</em></a></li>
</ul></li>
<li class="chapter" data-level="" data-path="about-the-author.html"><a href="about-the-author.html"><i class="fa fa-check"></i>About the Author</a></li>
<li class="divider"></li>
<li><a href="http://dyerlab.bio.vcu.edu" target="blank">the Dyer Laboratory</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Applied Population Genetics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="genetic-drift" class="section level1 imageChapter">
<h1><span class="header-section-number">8</span> Genetic Drift</h1>
<div class="chapter_image">
<img src="chapter_images/ch_beach_donkey.jpg">
</div>
<p>Genetic drift is the stochastic process resulting from a reduction in the size of a population. If populations are infinite in size, both allele and genotype frequencies should remain constant through time. However, if populations are not infinite (reality knocking), then drift may act as a diversifying force. This chapter examines the process of drift how we can measure its consequences.</p>
<p>Genetic drift is the process by which random selections of gametes produced during mating events result in stochastic changes in allele frequencies through time. This is not a terribly ground shattering idea, and our understanding of random chance provides some great insights into the underlying mechanisms operating in populations.</p>
<p>As an analogy, consider the simulation scenario where we have two alleles, <span class="math inline">\(A\)</span> and <span class="math inline">\(B\)</span>, at a haploid locus (for simplicity).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">Alleles &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;A&quot;</span>,<span class="st">&quot;B&quot;</span>)</code></pre></div>
<p>Collections are drawn at random using different sample sizes</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">sample_size &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">5</span>,<span class="dv">10</span>,<span class="dv">20</span>,<span class="dv">50</span>,<span class="dv">100</span>)</code></pre></div>
<p>We are implicitly assuming that the frequencies of these two alleles are identical (e.g., <span class="math inline">\(p_A = p_B = 0.5\)</span>). We can see the effects of genetic drift by running this ‘simulation’ in which populations are randomly created selected and used to estimate the frequency of an allele and how sample size influences the stability of the allele frequency. If <code>N</code> does not influence the estimation of allele frequencies then all examples should yield an estimated allele frequency equal to <span class="math inline">\(0.5\)</span> (as they truly are).</p>
<p>To do this, we set up the parameters in a data.frame</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">df &lt;-<span class="st"> </span><span class="kw">data.frame</span>( <span class="dt">N =</span> <span class="kw">rep</span>( sample_size, <span class="dt">each=</span><span class="dv">50</span> ),<span class="dt">FreqA =</span> <span class="ot">NA</span>)</code></pre></div>
<p>and then go through data frame, each row in order, and pull out random collections from Alleles in sizes equal to sample_size. To do this, I’m going to use a <code>for()</code> loop. This is simply a function that repeats a set of code a specified number of times. In this case, I will need to take the first row, make a random sample of alleles given the value of <code>df\$N[1]</code>, find the frequency of an allele in that random sample, and then store the frequency back into the data.frame at <code>df$FreqA[1]</code>. I then need to go to the second row and do the same thing again. And then the third row. Then the fourth. etc.</p>
<p>The format of a <code>for()</code> loop is pretty easy. You define a sequence of values to use (in this case I will use the variable named row to contain the sequence of integers from <code>1:nrow(df)</code>). Each time row is assigned a value, the code within the curly brackets ‘{}’ after the <code>for()</code> statement is executed and the value of row is constant during that iteration.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">for( row in <span class="dv">1</span>:<span class="kw">nrow</span>(df) ) {
  <span class="co"># make random selection from Alleles with replacement of a particular size</span>
  a &lt;-<span class="st"> </span><span class="kw">sample</span>( Alleles, <span class="dt">size=</span>df$N[row],<span class="dt">replace =</span> T)
  <span class="co"># find the frequency</span>
  f &lt;-<span class="st"> </span><span class="kw">sum</span>( a ==<span class="st"> &quot;A&quot;</span> ) /<span class="st"> </span><span class="kw">length</span>(a)
  <span class="co"># assign it back to the data.frame</span>
  df$FreqA[row] &lt;-<span class="st"> </span>f
}</code></pre></div>
<p>Here, row starts out at 1, then the code inside the brackets is run with <code>row=1</code>. After it reaches the end of the loop, row is incremented by one and the code is then run with <code>row=2</code>. This is repeated until <code>row = nrow(df)</code> and then the looping is finished. If you type this in and do it, you have done your first stochastic (some would call it Monte Carlo) simulation! Congratulations. That wasn’t so hard.</p>
<p>Lets now look at the data.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">summary</span>(df)</code></pre></div>
<pre><code>##        N           FreqA       
##  Min.   :  5   Min.   :0.1000  
##  1st Qu.: 10   1st Qu.:0.4125  
##  Median : 20   Median :0.5000  
##  Mean   : 37   Mean   :0.4942  
##  3rd Qu.: 50   3rd Qu.:0.5600  
##  Max.   :100   Max.   :1.0000</code></pre>
<p>The frequencies observed in each run are all over the place, basically covering the entire allowable range for an allele frequency. More importantly though lets look graphically at the allele frequencies using a boxplot where frequencies are grouped by sample size, <code>df$N</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(ggplot2)
p &lt;-<span class="st"> </span><span class="kw">ggplot</span>(df, <span class="kw">aes</span>(<span class="dt">x=</span><span class="kw">factor</span>(N), <span class="dt">y=</span>FreqA)) +<span class="st"> </span><span class="kw">geom_boxplot</span>() 
p +<span class="st"> </span><span class="kw">xlab</span>(<span class="st">&quot;Population Size&quot;</span>) +<span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;Estimated Allele Frequency&quot;</span>)</code></pre></div>
<div class="figure">
<img src="amg_files/figure-html/drift_simulation-1.png" alt="Allele frequencies estimated from random selections of alleles under varying sample sizes." width="672" />
<p class="caption">
(#fig:drift_simulation)Allele frequencies estimated from random selections of alleles under varying sample sizes.
</p>
</div>
<p>The plot of these data show an asymptotic reduction in the variation around the estimated allele frequency as you increase <code>N</code>.</p>
<p>This is to be expected, as we sample more individuals, we will become more confident that the parameters we estimate are closer to the real value. In statistics, this is the law of large numbers or the Central Limit Theorem. It is the same thing here. If there is a small population, each mating event may have the opportunity to have a wide fluctuation in gamete allele frequencies. All other things being equal, the frequencies will change each generation, an amount inversely proportional to the size of the population.</p>
<p>Small populations are not rare, some organisms exist only at small densities (e.g., top predators). Others may have undergone population bottlenecks, exist in marginal habitats at the edge of their tolerance range, or have a life-history that goes through periodic outbreaks.</p>
<p>Algebraically, we can determine the probability of observing a specific number of alleles in the next generation given random mating assumptions and a small sample size. The selection of diploid alleles follows a binomial expansion.</p>
<p><span class="math display">\[
P(N_A|N,p_A) = \frac{2N!}{N_A!(2N-N_A)!}p_A^{N_A}(1-p_A)^{2N-N_A}
\]</span></p>
<p>This expansion has two parts:</p>
<p>The binomial coefficient, <span class="math inline">\(\frac{2N!}{N_A!(2N-N_A)!}\)</span>, that determines the number of different ways we can get two alleles by random draw and have <span class="math inline">\(N_A\)</span> samples of the <span class="math inline">\(A\)</span> allele.</p>
<p>The probability, <span class="math inline">\(p_A^{N_A}(1-p_A)^{2N-N_A}\)</span>, of observing any each occurrence of the those samples based upon the frequency of the <span class="math inline">\(A\)</span> allele, <span class="math inline">\(p_A\)</span>, in the population.</p>
<div class="comment_warning">
<p>
Your ability to use the binomial expansion to solve this kind of problem relies upon the sample size, which becomes rather large rather quickly. In fact, R cannot estimate the value over 171!
</p>
</div>
<p>As an example, consider the idealized situation where we have <span class="math inline">\(N=20\)</span> individuals, all of which are heterozygotes. That means that for a diploid locus, <span class="math inline">\(N_A = N_B = 20\)</span>. The probability of observing exactly 20 <span class="math inline">\(A\)</span> alleles in the next generation can be calculated as:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">p &lt;-<span class="st"> </span><span class="fl">0.5</span>
N &lt;-<span class="st"> </span><span class="dv">20</span>
N.A &lt;-<span class="st"> </span><span class="dv">20</span></code></pre></div>
<p>where the binomial coefficient is given by</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">coef &lt;-<span class="st"> </span><span class="kw">factorial</span>(<span class="dv">2</span>*N)/<span class="st"> </span>(<span class="kw">factorial</span>(N.A)*<span class="kw">factorial</span>(<span class="dv">2</span>*N -<span class="st"> </span>N.A))
coef</code></pre></div>
<pre><code>## [1] 137846528820</code></pre>
<p>and the probability of each one of those is</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">prob &lt;-<span class="st"> </span>p^N.A *<span class="st"> </span>(<span class="dv">1</span> -<span class="st"> </span>p)^(<span class="dv">2</span>*N-N.A)
prob</code></pre></div>
<pre><code>## [1] 9.094947e-13</code></pre>
<p>and the final probability associated with observing <span class="math inline">\(N_A=20\)</span> after a single round of random mating is</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">Prob<span class="fl">.20</span>.A &lt;-<span class="st"> </span>coef*prob
Prob<span class="fl">.20</span>.A</code></pre></div>
<pre><code>## [1] 0.1253707</code></pre>
<p>Which is not so good, if you are interested in stability of allele frequencies. Only 12.5370688% of the time would the allele frequency stay the same and 87.4629312% of the time allele frequencies would change. This change is not so much an evolutionary forces as it is a stochastic process associated with low sample sizes.</p>
<p>So if it is not that likely that the allele frequencies will stay the same, what does the entire distribution of expected allele occurrences look like? Lets set the variable N.A equal to a sequence of potential allele counts (from 0 to 2*N) and then estimate and plot the distribution.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">N.A &lt;-<span class="st"> </span><span class="dv">0</span>:(<span class="dv">2</span>*N)
Frequency &lt;-<span class="st"> </span><span class="kw">factorial</span>(<span class="dv">2</span>*N)/( <span class="kw">factorial</span>(N.A)*<span class="kw">factorial</span>(<span class="dv">2</span>*N-N.A)) *<span class="st"> </span>p^N.A *<span class="st"> </span>(<span class="dv">1</span>-p)^(<span class="dv">2</span>*N-N.A)
df &lt;-<span class="st"> </span><span class="kw">data.frame</span>(N.A,Frequency)
<span class="kw">ggplot</span>(df,<span class="kw">aes</span>(<span class="dt">x=</span>N.A,<span class="dt">y=</span>Frequency)) +<span class="st"> </span><span class="kw">geom_bar</span>(<span class="dt">stat=</span><span class="st">&quot;identity&quot;</span>) +<span class="st"> </span><span class="kw">xlab</span>(<span class="kw">expression</span>(N[a]))</code></pre></div>
<p><img src="amg_files/figure-html/unnamed-chunk-216-1.png" width="672" /></p>
<p>Even for the tails of this distribution, the probabilities are not zero (here are the first 10 entries)</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">df[<span class="dv">1</span>:<span class="dv">10</span>,]</code></pre></div>
<pre><code>##    N.A    Frequency
## 1    0 9.094947e-13
## 2    1 3.637979e-11
## 3    2 7.094059e-10
## 4    3 8.985808e-09
## 5    4 8.311872e-08
## 6    5 5.984548e-07
## 7    6 3.490986e-06
## 8    7 1.695622e-05
## 9    8 6.994440e-05
## 10   9 2.486912e-04</code></pre>
<p>meaning that it is possible to start with everyone as a heterozygote and after a single random mating event, the population may either be fixed for the <span class="math inline">\(A\)</span> allele or have lost it altogether and only have aa homozygotes. The shape of this distribution is not only influenced by <span class="math inline">\(N\)</span> but also by <span class="math inline">\(p\)</span>. Here is what it looks like if the frequency of the allele is <span class="math inline">\(0.10\)</span> instead of <span class="math inline">\(0.50\)</span>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">p &lt;-<span class="st"> </span><span class="fl">0.1</span>
Frequency &lt;-<span class="st"> </span><span class="kw">factorial</span>(<span class="dv">2</span>*N)/( <span class="kw">factorial</span>(N.A)*<span class="kw">factorial</span>(<span class="dv">2</span>*N-N.A)) *<span class="st"> </span>p^N.A *<span class="st"> </span>(<span class="dv">1</span>-p)^(<span class="dv">2</span>*N-N.A)
df &lt;-<span class="st"> </span><span class="kw">data.frame</span>(N.A,Frequency)
<span class="kw">ggplot</span>(df,<span class="kw">aes</span>(<span class="dt">x=</span>N.A,<span class="dt">y=</span>Frequency)) +<span class="st"> </span><span class="kw">geom_bar</span>(<span class="dt">stat=</span><span class="st">&quot;identity&quot;</span>) +<span class="st"> </span><span class="kw">xlab</span>(<span class="kw">expression</span>(N[a])) </code></pre></div>
<p><img src="amg_files/figure-html/unnamed-chunk-218-1.png" width="672" /></p>
<div id="consequences-of-genetic-drift" class="section level2">
<h2><span class="header-section-number">8.1</span> Consequences of Genetic Drift</h2>
<p>The more skewed the allele frequency is away from <span class="math inline">\(0.5\)</span>, the more likely the population will be to become fixed for one allele or the other.</p>
<div id="time-to-fixation" class="section level3">
<h3><span class="header-section-number">8.1.1</span> Time to Fixation</h3>
<p>In fact, there is a well known relationship between the probability of fixation and the combination of both allele frequency and population size. Namely, the expected time to fixation, tfix (in generations), for a population of size Ne with two alleles (occurring p and q) is:</p>
<p><span class="math display">\[
t_{fix} = \frac{-4N_e(1-p)\log{(1-p})}{p}
\]</span></p>
<p>which for values of <span class="math inline">\(N\)</span> for <span class="math inline">\(p=c(0.01, 0.1, 0.25, 0.5)\)</span> are</p>
<p><img src="amg_files/figure-html/unnamed-chunk-219-1.png" width="672" /></p>
<p>This parameter Ne can be quite deviant from the census size <span class="math inline">\(N\)</span> depending upon several features of the organisms life history. We return to that later and discuss it in depth, for the time being, lets just assume it is a measure of the size of a population. That said, the stochastic selection of alleles due only to population size, can have significant effects on allele frequencies, available genetic diversity, and genotypic composition. In this simple two-allele system (often referred to as the Wright-Fisher model), if drift is the only feature that is influencing allele and genotype frequencies, the variance in allele frequencies through time has an expectation of:</p>
<p><span class="math display">\[
\sigma^2_p = pq\left[ 1-exp\left(-\frac{t}{2N_e}\right)\right]
\]</span></p>
<p>which if examined for changes in Ne for fixed p=0.5</p>
<div class="figure"><span id="fig:unnamed-chunk-220"></span>
<img src="amg_files/figure-html/unnamed-chunk-220-1.png" alt="Expected variance in allele frequencies through time for a Wright-Fisher model of genetic drift for three different effective population sizes." width="672" />
<p class="caption">
Figure 8.1: Expected variance in allele frequencies through time for a Wright-Fisher model of genetic drift for three different effective population sizes.
</p>
</div>
<p>or changes in p for fixed Ne = 100 are</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">df &lt;-<span class="st"> </span><span class="kw">data.frame</span>( <span class="dt">Generation=</span><span class="kw">c</span>(t,t,t),
                  <span class="dt">sigma_p=</span> <span class="kw">c</span>( <span class="kw">ne</span>(t,<span class="dv">100</span>,.<span class="dv">1</span>,.<span class="dv">9</span>), <span class="kw">ne</span>(t,<span class="dv">100</span>,.<span class="dv">25</span>,.<span class="dv">75</span>), <span class="kw">ne</span>(t,<span class="dv">100</span>,.<span class="dv">5</span>,.<span class="dv">5</span>) ),
                  <span class="dt">p=</span><span class="kw">rep</span>(<span class="kw">c</span>(<span class="st">&quot;p=0.10&quot;</span>,<span class="st">&quot;p=0.25&quot;</span>,<span class="st">&quot;p=0.50&quot;</span>), <span class="dt">each=</span><span class="dv">100</span>) ) 
<span class="kw">ggplot</span>( df, <span class="kw">aes</span>(<span class="dt">x=</span>Generation, <span class="dt">y=</span>sigma_p, <span class="dt">color=</span>p)) +<span class="st"> </span><span class="kw">geom_line</span>() +
<span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Generation (t)&quot;</span>) +<span class="st"> </span><span class="kw">ylab</span>(<span class="kw">expression</span>(sigma[p]^<span class="dv">2</span>))</code></pre></div>
<div class="figure"><span id="fig:unnamed-chunk-221"></span>
<img src="amg_files/figure-html/unnamed-chunk-221-1.png" alt="Expected variance in allele frequencies through time for a 2-allele Wright-Fisher model of genetic drift across different starting allele frequencies." width="672" />
<p class="caption">
Figure 8.2: Expected variance in allele frequencies through time for a 2-allele Wright-Fisher model of genetic drift across different starting allele frequencies.
</p>
</div>
<p>The important distinction here between these two graphs are that:</p>
<ol style="list-style-type: decimal">
<li>For different effective population sizes, the larger the population, the more stable the allele frequencies through time. For <span class="math inline">\(N_e = 1000\)</span>, the variance in allele frequencies is relatively small, compared to the other population sizes, even after 100 generations.<br />
</li>
<li>Allele frequencies show the opposite effect, with low allele frequencies, the variance is less than that in populations with larger allele frequencies (maximizing when <span class="math inline">\(p = \frac{1}{\ell}\)</span>, where <span class="math inline">\(\ell\)</span> is the number of alleles). There is more genetic variance with a more even distribution of allele frequencies (something we will return to when we talk about Fishers Fundamental Theorem).</li>
</ol>
</div>
<div id="time-to-allele-loss" class="section level3">
<h3><span class="header-section-number">8.1.2</span> Time to Allele Loss</h3>
<p>Time to fixation relates to the loss of all alleles, though in the data we often deal with, there is not a simple <span class="math inline">\(p=q\)</span>, two allele system. However, the models developed thus far, can give us an idea of the expected time to allele loss by rearranging the expectations a bit.</p>
<p><span class="math display">\[
t_{loss} = -\ln(p)\frac{4N_ep}{(1-p)}
\]</span></p>
</div>
</div>
<div id="mutation-drift" class="section level2">
<h2><span class="header-section-number">8.2</span> Mutation &amp; Drift</h2>
<p>If mutation is also a factor, then the time to allele loss, tloss, is modified a bit. We will come back to discussions of mutation in a later chapter and get a bit more in-depth to its consequences. However, depending upon the mutation model, its effects are generally that it increases the rate of allele loss rather than decrease it. This is because a random mutation (in a system where there are more than two alleles) is more likely to make allele A become something else than make one of the potential large number of other alleles into an A.</p>
<p>With a non-zero mutation rate, <span class="math inline">\(\mu\)</span>, the time to loss is approximated to be</p>
<p><span class="math display">\[
t_{loss} = \frac{1}{\mu}
\]</span></p>
<p>if and only if <span class="math inline">\(N_e\mu\)</span> is very small (e.g., &lt;&lt; 1). However, if it is larger than that, the expectation can be approximated as:</p>
<p><span class="math display">\[
t_{loss} = \frac{\ln(N_e\mu) + \gamma}{\mu}
\]</span></p>
<p>where the <span class="math inline">\(\gamma\)</span> term is Euler’s constant (Masel et al. 2007).</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="hardy-weinberg-equilibrium.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="effective-population-size.html" class="navigation navigation-next " aria-label="Next page""><i class="fa fa-angle-right"></i></a>

<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": false,
"twitter": false,
"google": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "weibo", "instapaper"],
"instapaper": false
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "section",
"scroll_highlight": true
},
"search": true
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    if (location.protocol !== "file:" && /^https?:/.test(script.src))
      script.src  = script.src.replace(/^https?:/, '');
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
