<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>Applied Population Genetics</title>
  <meta content="text/html; charset=UTF-8" http-equiv="Content-Type">
  <meta name="description" content="A textbook for the use of R in spatial genetic data analysis.">
  <meta name="generator" content="bookdown 0.3 and GitBook 2.6.7">

  <meta property="og:title" content="Applied Population Genetics" />
  <meta property="og:type" content="book" />
  
  <meta property="og:image" content="media/TitlePage_small.png" />
  <meta property="og:description" content="A textbook for the use of R in spatial genetic data analysis." />
  <meta name="github-repo" content="dyerlab/AMG" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Applied Population Genetics" />
  
  <meta name="twitter:description" content="A textbook for the use of R in spatial genetic data analysis." />
  <meta name="twitter:image" content="media/TitlePage_small.png" />

<meta name="author" content="Rodney J. Dyer">



  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="parentage-analyses.html">
<link rel="next" href="population-models.html">

<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />







<script src="libs/htmlwidgets-0.8/htmlwidgets.js"></script>
<link href="libs/leaflet-0.7.7/leaflet.css" rel="stylesheet" />
<script src="libs/leaflet-0.7.7/leaflet.js"></script>
<link href="libs/leafletfix-1.0.0/leafletfix.css" rel="stylesheet" />
<link href="libs/leaflet-label-0.2.2/leaflet.label.css" rel="stylesheet" />
<script src="libs/leaflet-label-0.2.2/leaflet.label.js"></script>
<script src="libs/Proj4Leaflet-0.7.2/proj4-compressed.js"></script>
<script src="libs/Proj4Leaflet-0.7.2/proj4leaflet.js"></script>
<script src="libs/leaflet-binding-1.1.0/leaflet.js"></script>
<script src="libs/leaflet-providers-1.0.27/leaflet-providers.js"></script>
<script src="libs/leaflet-providers-plugin-1.1.0/leaflet-providers-plugin.js"></script>
<script src="libs/d3-4.5.0/d3.min.js"></script>
<script src="libs/sankey-1/sankey.js"></script>
<script src="libs/sankeyNetwork-binding-0.3/sankeyNetwork.js"></script>
<script src="libs/dendroNetwork-binding-0.3/dendroNetwork.js"></script>
<script src="libs/simpleNetwork-binding-0.3/simpleNetwork.js"></script>
<script src="libs/forceNetwork-binding-0.3/forceNetwork.js"></script>


<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; background-color: #f8f8f8; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
pre, code { background-color: #f8f8f8; }
code > span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code > span.dt { color: #204a87; } /* DataType */
code > span.dv { color: #0000cf; } /* DecVal */
code > span.bn { color: #0000cf; } /* BaseN */
code > span.fl { color: #0000cf; } /* Float */
code > span.ch { color: #4e9a06; } /* Char */
code > span.st { color: #4e9a06; } /* String */
code > span.co { color: #8f5902; font-style: italic; } /* Comment */
code > span.ot { color: #8f5902; } /* Other */
code > span.al { color: #ef2929; } /* Alert */
code > span.fu { color: #000000; } /* Function */
code > span.er { color: #a40000; font-weight: bold; } /* Error */
code > span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #000000; } /* Constant */
code > span.sc { color: #000000; } /* SpecialChar */
code > span.vs { color: #4e9a06; } /* VerbatimString */
code > span.ss { color: #4e9a06; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #000000; } /* Variable */
code > span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code > span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code > span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code > span.ex { } /* Extension */
code > span.at { color: #c4a000; } /* Attribute */
code > span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code > span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
</style>

<link rel="stylesheet" href="media/style.css" type="text/css" />
</head>

<body>


  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Applied Population Genetics</a><ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#logistics"><i class="fa fa-check"></i>Logistics</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#motivation"><i class="fa fa-check"></i>Motivation</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#typography"><i class="fa fa-check"></i>Typography</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#interactive-content"><i class="fa fa-check"></i>Interactive Content</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#dedication"><i class="fa fa-check"></i>Dedication</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="the-r-ecosystem.html"><a href="the-r-ecosystem.html"><i class="fa fa-check"></i><b>1</b> The R Ecosystem</a><ul>
<li class="chapter" data-level="1.1" data-path="the-r-ecosystem.html"><a href="the-r-ecosystem.html#getting-r-configured"><i class="fa fa-check"></i><b>1.1</b> Getting R Configured</a><ul>
<li class="chapter" data-level="1.1.1" data-path="the-r-ecosystem.html"><a href="the-r-ecosystem.html#packages"><i class="fa fa-check"></i><b>1.1.1</b> Packages</a></li>
<li class="chapter" data-level="1.1.2" data-path="the-r-ecosystem.html"><a href="the-r-ecosystem.html#cran"><i class="fa fa-check"></i><b>1.1.2</b> CRAN</a></li>
<li class="chapter" data-level="1.1.3" data-path="the-r-ecosystem.html"><a href="the-r-ecosystem.html#github"><i class="fa fa-check"></i><b>1.1.3</b> GitHub</a></li>
<li class="chapter" data-level="1.1.4" data-path="the-r-ecosystem.html"><a href="the-r-ecosystem.html#bioconductor"><i class="fa fa-check"></i><b>1.1.4</b> Bioconductor</a></li>
<li class="chapter" data-level="1.1.5" data-path="the-r-ecosystem.html"><a href="the-r-ecosystem.html#troublesome-packages"><i class="fa fa-check"></i><b>1.1.5</b> Troublesome Packages</a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="the-r-ecosystem.html"><a href="the-r-ecosystem.html#libraries-used-in-text"><i class="fa fa-check"></i><b>1.2</b> Libraries Used in Text</a></li>
<li class="chapter" data-level="1.3" data-path="the-r-ecosystem.html"><a href="the-r-ecosystem.html#the-rstudio-environment"><i class="fa fa-check"></i><b>1.3</b> The RStudio Environment</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="data-types.html"><a href="data-types.html"><i class="fa fa-check"></i><b>2</b> Data Types</a><ul>
<li class="chapter" data-level="2.1" data-path="data-types.html"><a href="data-types.html#numeric-data-types"><i class="fa fa-check"></i><b>2.1</b> Numeric Data Types</a><ul>
<li class="chapter" data-level="2.1.1" data-path="data-types.html"><a href="data-types.html#coercion-to-numeric"><i class="fa fa-check"></i><b>2.1.1</b> Coercion to Numeric</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="data-types.html"><a href="data-types.html#characters"><i class="fa fa-check"></i><b>2.2</b> Characters</a><ul>
<li class="chapter" data-level="2.2.1" data-path="data-types.html"><a href="data-types.html#concatenation-of-characters"><i class="fa fa-check"></i><b>2.2.1</b> Concatenation of Characters</a></li>
<li class="chapter" data-level="2.2.2" data-path="data-types.html"><a href="data-types.html#coercion-to-characters"><i class="fa fa-check"></i><b>2.2.2</b> Coercion to Characters</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="data-types.html"><a href="data-types.html#factors"><i class="fa fa-check"></i><b>2.3</b> Factors</a><ul>
<li class="chapter" data-level="2.3.1" data-path="data-types.html"><a href="data-types.html#missing-levels-in-factors"><i class="fa fa-check"></i><b>2.3.1</b> Missing Levels in Factors</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="data-types.html"><a href="data-types.html#logical-types"><i class="fa fa-check"></i><b>2.4</b> Logical Types</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="data-containers.html"><a href="data-containers.html"><i class="fa fa-check"></i><b>3</b> Data Containers</a><ul>
<li class="chapter" data-level="3.1" data-path="data-containers.html"><a href="data-containers.html#vectors"><i class="fa fa-check"></i><b>3.1</b> Vectors</a></li>
<li class="chapter" data-level="3.2" data-path="data-containers.html"><a href="data-containers.html#matrices"><i class="fa fa-check"></i><b>3.2</b> Matrices</a></li>
<li class="chapter" data-level="3.3" data-path="data-containers.html"><a href="data-containers.html#lists"><i class="fa fa-check"></i><b>3.3</b> Lists</a></li>
<li class="chapter" data-level="3.4" data-path="data-containers.html"><a href="data-containers.html#data-frames"><i class="fa fa-check"></i><b>3.4</b> Data Frames</a><ul>
<li class="chapter" data-level="3.4.1" data-path="data-containers.html"><a href="data-containers.html#indexing-data-frames"><i class="fa fa-check"></i><b>3.4.1</b> Indexing Data Frames</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="programming.html"><a href="programming.html"><i class="fa fa-check"></i><b>4</b> Programming</a><ul>
<li class="chapter" data-level="4.1" data-path="programming.html"><a href="programming.html#function-writing"><i class="fa fa-check"></i><b>4.1</b> Function Writing</a></li>
<li class="chapter" data-level="4.2" data-path="programming.html"><a href="programming.html#variable-scope"><i class="fa fa-check"></i><b>4.2</b> Variable Scope</a></li>
<li class="chapter" data-level="4.3" data-path="programming.html"><a href="programming.html#decision-making"><i class="fa fa-check"></i><b>4.3</b> Decision Making</a><ul>
<li class="chapter" data-level="4.3.1" data-path="programming.html"><a href="programming.html#the-if-pattern"><i class="fa fa-check"></i><b>4.3.1</b> The if Pattern</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="programming.html"><a href="programming.html#the-if-else-pattern"><i class="fa fa-check"></i><b>4.4</b> The <code>if-else</code> Pattern</a></li>
<li class="chapter" data-level="4.5" data-path="programming.html"><a href="programming.html#flow-control"><i class="fa fa-check"></i><b>4.5</b> Flow Control</a><ul>
<li class="chapter" data-level="4.5.1" data-path="programming.html"><a href="programming.html#the-for-loop"><i class="fa fa-check"></i><b>4.5.1</b> The for() Loop</a></li>
<li class="chapter" data-level="4.5.2" data-path="programming.html"><a href="programming.html#short-circuiting-the-loop"><i class="fa fa-check"></i><b>4.5.2</b> Short Circuiting the Loop</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="graphics.html"><a href="graphics.html"><i class="fa fa-check"></i><b>5</b> Graphics</a><ul>
<li class="chapter" data-level="5.1" data-path="graphics.html"><a href="graphics.html#univariate-data"><i class="fa fa-check"></i><b>5.1</b> Univariate Data</a></li>
<li class="chapter" data-level="5.2" data-path="graphics.html"><a href="graphics.html#plotting-parameters"><i class="fa fa-check"></i><b>5.2</b> Plotting Parameters</a></li>
<li class="chapter" data-level="5.3" data-path="graphics.html"><a href="graphics.html#density-plots"><i class="fa fa-check"></i><b>5.3</b> Density Plots</a></li>
<li class="chapter" data-level="5.4" data-path="graphics.html"><a href="graphics.html#overlaying-plots"><i class="fa fa-check"></i><b>5.4</b> Overlaying Plots</a></li>
<li class="chapter" data-level="5.5" data-path="graphics.html"><a href="graphics.html#boxplots"><i class="fa fa-check"></i><b>5.5</b> Boxplots</a></li>
<li class="chapter" data-level="5.6" data-path="graphics.html"><a href="graphics.html#scatter-plots"><i class="fa fa-check"></i><b>5.6</b> Scatter Plots</a></li>
<li class="chapter" data-level="5.7" data-path="graphics.html"><a href="graphics.html#multiple-plots"><i class="fa fa-check"></i><b>5.7</b> Multiple Plots</a></li>
<li class="chapter" data-level="5.8" data-path="graphics.html"><a href="graphics.html#color-palettes"><i class="fa fa-check"></i><b>5.8</b> Color Palettes</a></li>
<li class="chapter" data-level="5.9" data-path="graphics.html"><a href="graphics.html#rcolorbrewer"><i class="fa fa-check"></i><b>5.9</b> RColorBrewer</a></li>
<li class="chapter" data-level="5.10" data-path="graphics.html"><a href="graphics.html#saving-imagery"><i class="fa fa-check"></i><b>5.10</b> Saving Imagery</a></li>
<li class="chapter" data-level="5.11" data-path="graphics.html"><a href="graphics.html#interactive-graphics"><i class="fa fa-check"></i><b>5.11</b> Interactive Graphics</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="reproducable-research.html"><a href="reproducable-research.html"><i class="fa fa-check"></i><b>6</b> Reproducable Research</a><ul>
<li class="chapter" data-level="6.1" data-path="reproducable-research.html"><a href="reproducable-research.html#keeping-data-on-google-drive"><i class="fa fa-check"></i><b>6.1</b> Keeping Data on Google Drive</a></li>
<li class="chapter" data-level="6.2" data-path="reproducable-research.html"><a href="reproducable-research.html#integrating-into-slack"><i class="fa fa-check"></i><b>6.2</b> Integrating into Slack</a><ul>
<li class="chapter" data-level="6.2.1" data-path="reproducable-research.html"><a href="reproducable-research.html#fixing-problem-with-version-1.4.1"><i class="fa fa-check"></i><b>6.2.1</b> Fixing Problem with Version 1.4.1</a></li>
<li class="chapter" data-level="6.2.2" data-path="reproducable-research.html"><a href="reproducable-research.html#configuration"><i class="fa fa-check"></i><b>6.2.2</b> Configuration</a></li>
<li class="chapter" data-level="6.2.3" data-path="reproducable-research.html"><a href="reproducable-research.html#sending-text-and-graphical-ouput"><i class="fa fa-check"></i><b>6.2.3</b> Sending Text and Graphical Ouput</a></li>
<li class="chapter" data-level="6.2.4" data-path="reproducable-research.html"><a href="reproducable-research.html#markdown"><i class="fa fa-check"></i><b>6.2.4</b> Markdown</a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>I Single Population</b></span></li>
<li class="chapter" data-level="7" data-path="hardy-weinberg-equilibrium.html"><a href="hardy-weinberg-equilibrium.html"><i class="fa fa-check"></i><b>7</b> Hardy-Weinberg Equilibrium</a><ul>
<li class="chapter" data-level="7.1" data-path="hardy-weinberg-equilibrium.html"><a href="hardy-weinberg-equilibrium.html#genetic-assumptions"><i class="fa fa-check"></i><b>7.1</b> Genetic Assumptions</a><ul>
<li class="chapter" data-level="7.1.1" data-path="hardy-weinberg-equilibrium.html"><a href="hardy-weinberg-equilibrium.html#punnet-squares"><i class="fa fa-check"></i><b>7.1.1</b> Punnet Squares</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="hardy-weinberg-equilibrium.html"><a href="hardy-weinberg-equilibrium.html#evolutionary-assumptions"><i class="fa fa-check"></i><b>7.2</b> Evolutionary Assumptions</a></li>
<li class="chapter" data-level="7.3" data-path="hardy-weinberg-equilibrium.html"><a href="hardy-weinberg-equilibrium.html#the-mechanics"><i class="fa fa-check"></i><b>7.3</b> The Mechanics</a></li>
<li class="chapter" data-level="7.4" data-path="hardy-weinberg-equilibrium.html"><a href="hardy-weinberg-equilibrium.html#trans-generational-transition-probabilities"><i class="fa fa-check"></i><b>7.4</b> Trans-Generational Transition Probabilities</a></li>
<li class="chapter" data-level="7.5" data-path="hardy-weinberg-equilibrium.html"><a href="hardy-weinberg-equilibrium.html#consequences-of-hwe"><i class="fa fa-check"></i><b>7.5</b> Consequences of HWE</a></li>
<li class="chapter" data-level="7.6" data-path="hardy-weinberg-equilibrium.html"><a href="hardy-weinberg-equilibrium.html#zygotic-gene-flow"><i class="fa fa-check"></i><b>7.6</b> Zygotic Gene Flow</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="genetic-drift.html"><a href="genetic-drift.html"><i class="fa fa-check"></i><b>8</b> Genetic Drift</a><ul>
<li class="chapter" data-level="8.1" data-path="genetic-drift.html"><a href="genetic-drift.html#consequences-of-genetic-drift"><i class="fa fa-check"></i><b>8.1</b> Consequences of Genetic Drift</a><ul>
<li class="chapter" data-level="8.1.1" data-path="genetic-drift.html"><a href="genetic-drift.html#time-to-fixation"><i class="fa fa-check"></i><b>8.1.1</b> Time to Fixation</a></li>
<li class="chapter" data-level="8.1.2" data-path="genetic-drift.html"><a href="genetic-drift.html#time-to-allele-loss"><i class="fa fa-check"></i><b>8.1.2</b> Time to Allele Loss</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="genetic-drift.html"><a href="genetic-drift.html#mutation-drift"><i class="fa fa-check"></i><b>8.2</b> Mutation &amp; Drift</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="effective-population-size.html"><a href="effective-population-size.html"><i class="fa fa-check"></i><b>9</b> Effective Population Size</a><ul>
<li class="chapter" data-level="9.1" data-path="effective-population-size.html"><a href="effective-population-size.html#demographic-effective-population-sizes"><i class="fa fa-check"></i><b>9.1</b> Demographic Effective Population Sizes</a></li>
<li class="chapter" data-level="9.2" data-path="effective-population-size.html"><a href="effective-population-size.html#genetic-effective-population-size"><i class="fa fa-check"></i><b>9.2</b> Genetic Effective Population Size</a><ul>
<li class="chapter" data-level="9.2.1" data-path="effective-population-size.html"><a href="effective-population-size.html#inbreeding-effective-population-size"><i class="fa fa-check"></i><b>9.2.1</b> Inbreeding Effective Population Size</a></li>
<li class="chapter" data-level="9.2.2" data-path="effective-population-size.html"><a href="effective-population-size.html#variance-effective-population-size"><i class="fa fa-check"></i><b>9.2.2</b> Variance Effective Population Size</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="effective-population-size.html"><a href="effective-population-size.html#fluctuating-population-size"><i class="fa fa-check"></i><b>9.3</b> Fluctuating Population Size</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="mutation.html"><a href="mutation.html"><i class="fa fa-check"></i><b>10</b> Mutation</a><ul>
<li class="chapter" data-level="10.1" data-path="mutation.html"><a href="mutation.html#mutation-models"><i class="fa fa-check"></i><b>10.1</b> Mutation Models</a><ul>
<li class="chapter" data-level="10.1.1" data-path="mutation.html"><a href="mutation.html#fixed-allele-model"><i class="fa fa-check"></i><b>10.1.1</b> Fixed Allele Model</a></li>
<li class="chapter" data-level="10.1.2" data-path="mutation.html"><a href="mutation.html#infinite-allele-model"><i class="fa fa-check"></i><b>10.1.2</b> Infinite Allele Model</a></li>
<li class="chapter" data-level="10.1.3" data-path="mutation.html"><a href="mutation.html#stepwise-mutation-model"><i class="fa fa-check"></i><b>10.1.3</b> Stepwise Mutation Model</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="mutation.html"><a href="mutation.html#mutation-and-inbreeding"><i class="fa fa-check"></i><b>10.2</b> Mutation and Inbreeding</a></li>
<li class="chapter" data-level="10.3" data-path="mutation.html"><a href="mutation.html#estimating-mutation"><i class="fa fa-check"></i><b>10.3</b> Estimating Mutation</a><ul>
<li class="chapter" data-level="10.3.1" data-path="mutation.html"><a href="mutation.html#phenotypes"><i class="fa fa-check"></i><b>10.3.1</b> Phenotypes</a></li>
<li class="chapter" data-level="10.3.2" data-path="mutation.html"><a href="mutation.html#genotypes"><i class="fa fa-check"></i><b>10.3.2</b> Genotypes</a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>II Nonrandom Rating</b></span></li>
<li class="chapter" data-level="11" data-path="selfing.html"><a href="selfing.html"><i class="fa fa-check"></i><b>11</b> Selfing</a><ul>
<li class="chapter" data-level="11.1" data-path="selfing.html"><a href="selfing.html#changes-in-f"><i class="fa fa-check"></i><b>11.1</b> Changes in F</a><ul>
<li class="chapter" data-level="11.1.1" data-path="selfing.html"><a href="selfing.html#theoretical-expectations"><i class="fa fa-check"></i><b>11.1.1</b> Theoretical Expectations</a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="selfing.html"><a href="selfing.html#simulation-example"><i class="fa fa-check"></i><b>11.2</b> Simulation Example</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="mixed-mating-systems.html"><a href="mixed-mating-systems.html"><i class="fa fa-check"></i><b>12</b> Mixed Mating Systems</a><ul>
<li class="chapter" data-level="12.1" data-path="mixed-mating-systems.html"><a href="mixed-mating-systems.html#mixed-mating-genotype-frequency-equilibriums"><i class="fa fa-check"></i><b>12.1</b> Mixed Mating Genotype Frequency Equilibriums</a></li>
<li class="chapter" data-level="12.2" data-path="mixed-mating-systems.html"><a href="mixed-mating-systems.html#estimating-selfing-rates"><i class="fa fa-check"></i><b>12.2</b> Estimating Selfing Rates</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="wahlund-effects.html"><a href="wahlund-effects.html"><i class="fa fa-check"></i><b>13</b> Wahlund Effects</a></li>
<li class="part"><span><b>III Parent Offspring Data</b></span></li>
<li class="chapter" data-level="14" data-path="pedigrees.html"><a href="pedigrees.html"><i class="fa fa-check"></i><b>14</b> Pedigrees</a><ul>
<li class="chapter" data-level="14.1" data-path="pedigrees.html"><a href="pedigrees.html#expectations-for-f-in-a-pedigree."><i class="fa fa-check"></i><b>14.1</b> Expectations for F in a pedigree.</a></li>
<li class="chapter" data-level="14.2" data-path="pedigrees.html"><a href="pedigrees.html#drawing-pedigrees"><i class="fa fa-check"></i><b>14.2</b> Drawing Pedigrees</a></li>
<li class="chapter" data-level="14.3" data-path="pedigrees.html"><a href="pedigrees.html#skills"><i class="fa fa-check"></i><b>14.3</b> Skills</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="parentage-analyses.html"><a href="parentage-analyses.html"><i class="fa fa-check"></i><b>15</b> Parentage Analyses</a><ul>
<li class="chapter" data-level="15.1" data-path="parentage-analyses.html"><a href="parentage-analyses.html#paternity-vs.maternity"><i class="fa fa-check"></i><b>15.1</b> Paternity vs. Maternity</a></li>
<li class="chapter" data-level="15.2" data-path="parentage-analyses.html"><a href="parentage-analyses.html#probability-of-exclusion"><i class="fa fa-check"></i><b>15.2</b> Probability of Exclusion</a></li>
<li class="chapter" data-level="15.3" data-path="parentage-analyses.html"><a href="parentage-analyses.html#paternity-exclusion"><i class="fa fa-check"></i><b>15.3</b> Paternity Exclusion</a></li>
<li class="chapter" data-level="15.4" data-path="parentage-analyses.html"><a href="parentage-analyses.html#fractional-paternity"><i class="fa fa-check"></i><b>15.4</b> Fractional Paternity</a></li>
<li class="chapter" data-level="15.5" data-path="parentage-analyses.html"><a href="parentage-analyses.html#dispersal-kernels"><i class="fa fa-check"></i><b>15.5</b> Dispersal Kernels</a></li>
</ul></li>
<li class="part"><span><b>IV Subdivided Populations</b></span></li>
<li class="chapter" data-level="16" data-path="mapping-populations.html"><a href="mapping-populations.html"><i class="fa fa-check"></i><b>16</b> Mapping Populations</a><ul>
<li class="chapter" data-level="16.1" data-path="mapping-populations.html"><a href="mapping-populations.html#procedure"><i class="fa fa-check"></i><b>16.1</b> Procedure</a></li>
<li class="chapter" data-level="16.2" data-path="mapping-populations.html"><a href="mapping-populations.html#plotting-population-locations"><i class="fa fa-check"></i><b>16.2</b> Plotting Population Locations</a></li>
<li class="chapter" data-level="16.3" data-path="mapping-populations.html"><a href="mapping-populations.html#population-allele-frequencies"><i class="fa fa-check"></i><b>16.3</b> Population Allele Frequencies</a></li>
<li class="chapter" data-level="16.4" data-path="mapping-populations.html"><a href="mapping-populations.html#dynamic-maps"><i class="fa fa-check"></i><b>16.4</b> Dynamic maps</a><ul>
<li class="chapter" data-level="16.4.1" data-path="mapping-populations.html"><a href="mapping-populations.html#sankey-networks"><i class="fa fa-check"></i><b>16.4.1</b> Sankey Networks</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="17" data-path="population-models.html"><a href="population-models.html"><i class="fa fa-check"></i><b>17</b> Population Models</a><ul>
<li class="chapter" data-level="17.1" data-path="population-models.html"><a href="population-models.html#models-of-partition"><i class="fa fa-check"></i><b>17.1</b> Models of Partition</a></li>
<li class="chapter" data-level="17.2" data-path="population-models.html"><a href="population-models.html#island-mainland-model"><i class="fa fa-check"></i><b>17.2</b> Island Mainland Model</a></li>
<li class="chapter" data-level="17.3" data-path="population-models.html"><a href="population-models.html#the-island-model"><i class="fa fa-check"></i><b>17.3</b> The Island Model</a></li>
<li class="chapter" data-level="17.4" data-path="population-models.html"><a href="population-models.html#stepping-stone-models"><i class="fa fa-check"></i><b>17.4</b> Stepping Stone Models</a></li>
<li class="chapter" data-level="17.5" data-path="population-models.html"><a href="population-models.html#the-general-model"><i class="fa fa-check"></i><b>17.5</b> The General Model</a></li>
<li class="chapter" data-level="17.6" data-path="population-models.html"><a href="population-models.html#general-strategies"><i class="fa fa-check"></i><b>17.6</b> General Strategies</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="genetic-diversity.html"><a href="genetic-diversity.html"><i class="fa fa-check"></i><b>18</b> Genetic Diversity</a><ul>
<li class="chapter" data-level="18.1" data-path="genetic-diversity.html"><a href="genetic-diversity.html#allelic-diversity"><i class="fa fa-check"></i><b>18.1</b> Allelic Diversity</a></li>
<li class="chapter" data-level="18.2" data-path="genetic-diversity.html"><a href="genetic-diversity.html#genotypic-diversity"><i class="fa fa-check"></i><b>18.2</b> Genotypic Diversity</a></li>
<li class="chapter" data-level="18.3" data-path="genetic-diversity.html"><a href="genetic-diversity.html#samples-from-several-locales"><i class="fa fa-check"></i><b>18.3</b> Samples from Several Locales</a></li>
<li class="chapter" data-level="18.4" data-path="genetic-diversity.html"><a href="genetic-diversity.html#multilocus-diversity"><i class="fa fa-check"></i><b>18.4</b> Multilocus Diversity</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="rarefaction.html"><a href="rarefaction.html"><i class="fa fa-check"></i><b>19</b> Rarefaction</a><ul>
<li class="chapter" data-level="19.1" data-path="rarefaction.html"><a href="rarefaction.html#mapping-diversity"><i class="fa fa-check"></i><b>19.1</b> Mapping Diversity</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="genetic-distances.html"><a href="genetic-distances.html"><i class="fa fa-check"></i><b>20</b> Genetic Distances</a><ul>
<li class="chapter" data-level="20.1" data-path="genetic-distances.html"><a href="genetic-distances.html#individual-genetic-distance"><i class="fa fa-check"></i><b>20.1</b> Individual Genetic Distance</a><ul>
<li class="chapter" data-level="20.1.1" data-path="genetic-distances.html"><a href="genetic-distances.html#euclidean-distance"><i class="fa fa-check"></i><b>20.1.1</b> Euclidean Distance</a></li>
<li class="chapter" data-level="20.1.2" data-path="genetic-distances.html"><a href="genetic-distances.html#amova-distance"><i class="fa fa-check"></i><b>20.1.2</b> AMOVA Distance</a></li>
<li class="chapter" data-level="20.1.3" data-path="genetic-distances.html"><a href="genetic-distances.html#bray-curtis-individual-distance"><i class="fa fa-check"></i><b>20.1.3</b> Bray-Curtis Individual Distance</a></li>
<li class="chapter" data-level="20.1.4" data-path="genetic-distances.html"><a href="genetic-distances.html#ladder-distance"><i class="fa fa-check"></i><b>20.1.4</b> Ladder Distance</a></li>
</ul></li>
<li class="chapter" data-level="20.2" data-path="genetic-distances.html"><a href="genetic-distances.html#population-level-genetic-distances"><i class="fa fa-check"></i><b>20.2</b> Population-Level Genetic Distances</a><ul>
<li class="chapter" data-level="20.2.1" data-path="genetic-distances.html"><a href="genetic-distances.html#czekanowski-manhattan-distance"><i class="fa fa-check"></i><b>20.2.1</b> Czekanowski (Manhattan) Distance</a></li>
<li class="chapter" data-level="20.2.2" data-path="genetic-distances.html"><a href="genetic-distances.html#rogers-euclidean-distance"><i class="fa fa-check"></i><b>20.2.2</b> Rogers (Euclidean) Distance</a></li>
<li class="chapter" data-level="20.2.3" data-path="genetic-distances.html"><a href="genetic-distances.html#neis-genetic-distance"><i class="fa fa-check"></i><b>20.2.3</b> Nei’s Genetic Distance</a></li>
<li class="chapter" data-level="20.2.4" data-path="genetic-distances.html"><a href="genetic-distances.html#conditional-genetic-distance"><i class="fa fa-check"></i><b>20.2.4</b> Conditional Genetic Distance</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="21" data-path="genetic-structure.html"><a href="genetic-structure.html"><i class="fa fa-check"></i><b>21</b> Genetic Structure</a><ul>
<li class="chapter" data-level="21.1" data-path="genetic-structure.html"><a href="genetic-structure.html#f-statistics-and-thier-ilk"><i class="fa fa-check"></i><b>21.1</b> F-Statistics and Thier Ilk</a><ul>
<li class="chapter" data-level="21.1.1" data-path="genetic-structure.html"><a href="genetic-structure.html#issues-with-fixation-indices"><i class="fa fa-check"></i><b>21.1.1</b> Issues with Fixation Indices</a></li>
<li class="chapter" data-level="21.1.2" data-path="genetic-structure.html"><a href="genetic-structure.html#additional-x_st-like-parameters"><i class="fa fa-check"></i><b>21.1.2</b> Additional <span class="math inline">\(X_{ST}\)</span>-like Parameters</a></li>
</ul></li>
<li class="chapter" data-level="21.2" data-path="genetic-structure.html"><a href="genetic-structure.html#statistical-structure"><i class="fa fa-check"></i><b>21.2</b> Statistical Structure</a></li>
<li class="chapter" data-level="21.3" data-path="genetic-structure.html"><a href="genetic-structure.html#final-thoughts-on-structure"><i class="fa fa-check"></i><b>21.3</b> Final Thoughts on Structure</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="population-assignment.html"><a href="population-assignment.html"><i class="fa fa-check"></i><b>22</b> Population Assignment</a></li>
<li class="chapter" data-level="23" data-path="admixture.html"><a href="admixture.html"><i class="fa fa-check"></i><b>23</b> Admixture</a></li>
<li class="part"><span><b>V Statistical Models</b></span></li>
<li class="chapter" data-level="24" data-path="ordination-of-genetic-data.html"><a href="ordination-of-genetic-data.html"><i class="fa fa-check"></i><b>24</b> Ordination of Genetic Data</a><ul>
<li class="chapter" data-level="24.1" data-path="ordination-of-genetic-data.html"><a href="ordination-of-genetic-data.html#hierarchical-clustering"><i class="fa fa-check"></i><b>24.1</b> Hierarchical Clustering</a></li>
<li class="chapter" data-level="24.2" data-path="ordination-of-genetic-data.html"><a href="ordination-of-genetic-data.html#interactive-tree-display"><i class="fa fa-check"></i><b>24.2</b> Interactive Tree Display</a></li>
</ul></li>
<li class="chapter" data-level="25" data-path="distance-matrices.html"><a href="distance-matrices.html"><i class="fa fa-check"></i><b>25</b> Distance Matrices</a><ul>
<li class="chapter" data-level="25.1" data-path="distance-matrices.html"><a href="distance-matrices.html#visual-comparisons"><i class="fa fa-check"></i><b>25.1</b> Visual Comparisons</a></li>
<li class="chapter" data-level="25.2" data-path="distance-matrices.html"><a href="distance-matrices.html#the-mantel-test"><i class="fa fa-check"></i><b>25.2</b> The Mantel Test</a></li>
</ul></li>
<li class="chapter" data-level="26" data-path="linear-models.html"><a href="linear-models.html"><i class="fa fa-check"></i><b>26</b> Linear Models</a><ul>
<li class="chapter" data-level="26.1" data-path="linear-models.html"><a href="linear-models.html#dependent-independent-relationships"><i class="fa fa-check"></i><b>26.1</b> Dependent &amp; Independent Relationships</a></li>
</ul></li>
<li class="chapter" data-level="27" data-path="distance-networks.html"><a href="distance-networks.html"><i class="fa fa-check"></i><b>27</b> Distance Networks</a><ul>
<li class="chapter" data-level="27.1" data-path="distance-networks.html"><a href="distance-networks.html#graph-structure"><i class="fa fa-check"></i><b>27.1</b> Graph Structure</a><ul>
<li class="chapter" data-level="27.1.1" data-path="distance-networks.html"><a href="distance-networks.html#node-edge-attributes"><i class="fa fa-check"></i><b>27.1.1</b> Node &amp; Edge Attributes</a></li>
</ul></li>
<li class="chapter" data-level="27.2" data-path="distance-networks.html"><a href="distance-networks.html#plotting-a-graphs"><i class="fa fa-check"></i><b>27.2</b> Plotting a Graphs</a></li>
<li class="chapter" data-level="27.3" data-path="distance-networks.html"><a href="distance-networks.html#mapping-networks"><i class="fa fa-check"></i><b>27.3</b> Mapping Networks</a></li>
<li class="chapter" data-level="27.4" data-path="distance-networks.html"><a href="distance-networks.html#genetic-distance-graphs"><i class="fa fa-check"></i><b>27.4</b> Genetic Distance Graphs</a></li>
</ul></li>
<li class="chapter" data-level="28" data-path="population-graphs.html"><a href="population-graphs.html"><i class="fa fa-check"></i><b>28</b> Population Graphs</a><ul>
<li class="chapter" data-level="28.1" data-path="population-graphs.html"><a href="population-graphs.html#adding-data-to-a-graph"><i class="fa fa-check"></i><b>28.1</b> Adding data to a graph</a><ul>
<li class="chapter" data-level="28.1.1" data-path="population-graphs.html"><a href="population-graphs.html#plotting-a-graph-using-ggplot2-routines"><i class="fa fa-check"></i><b>28.1.1</b> Plotting a graph using <strong>ggplot2</strong> routines</a></li>
<li class="chapter" data-level="28.1.2" data-path="population-graphs.html"><a href="population-graphs.html#reading-existing-popgraph-files"><i class="fa fa-check"></i><b>28.1.2</b> Reading Existing <strong>popgraph</strong> Files</a></li>
<li class="chapter" data-level="28.1.3" data-path="population-graphs.html"><a href="population-graphs.html#saving-population-graph-objects"><i class="fa fa-check"></i><b>28.1.3</b> Saving Population Graph Objects</a></li>
</ul></li>
<li class="chapter" data-level="28.2" data-path="population-graphs.html"><a href="population-graphs.html#interactive-networks"><i class="fa fa-check"></i><b>28.2</b> Interactive Networks</a></li>
<li class="chapter" data-level="28.3" data-path="population-graphs.html"><a href="population-graphs.html#spatial-population-graphs"><i class="fa fa-check"></i><b>28.3</b> Spatial Population Graphs</a><ul>
<li class="chapter" data-level="28.3.1" data-path="population-graphs.html"><a href="population-graphs.html#integrating-google-and-ggplot2-for-plotting"><i class="fa fa-check"></i><b>28.3.1</b> Integrating Google and <strong>ggplot2</strong> for Plotting</a></li>
<li class="chapter" data-level="28.3.2" data-path="population-graphs.html"><a href="population-graphs.html#integrating-raster-maps"><i class="fa fa-check"></i><b>28.3.2</b> Integrating Raster Maps</a></li>
</ul></li>
<li class="chapter" data-level="28.4" data-path="population-graphs.html"><a href="population-graphs.html#extracting-spatial-data-using-population-graphs"><i class="fa fa-check"></i><b>28.4</b> Extracting Spatial Data Using Population Graphs</a><ul>
<li class="chapter" data-level="28.4.1" data-path="population-graphs.html"><a href="population-graphs.html#node-specific-data"><i class="fa fa-check"></i><b>28.4.1</b> Node Specific Data</a></li>
<li class="chapter" data-level="28.4.2" data-path="population-graphs.html"><a href="population-graphs.html#extracting-data-along-popgraph-edges"><i class="fa fa-check"></i><b>28.4.2</b> Extracting Data Along Popgraph Edges</a></li>
</ul></li>
<li class="chapter" data-level="28.5" data-path="population-graphs.html"><a href="population-graphs.html#extracting-graph-theoretic-parameters"><i class="fa fa-check"></i><b>28.5</b> Extracting Graph-Theoretic Parameters</a><ul>
<li class="chapter" data-level="28.5.1" data-path="population-graphs.html"><a href="population-graphs.html#matrix-representations-of-population-graph-objects"><i class="fa fa-check"></i><b>28.5.1</b> Matrix Representations of Population Graph Objects</a></li>
<li class="chapter" data-level="28.5.2" data-path="population-graphs.html"><a href="population-graphs.html#node-specific-parameters"><i class="fa fa-check"></i><b>28.5.2</b> Node Specific Parameters</a></li>
<li class="chapter" data-level="28.5.3" data-path="population-graphs.html"><a href="population-graphs.html#edge-specific-parameters"><i class="fa fa-check"></i><b>28.5.3</b> Edge Specific Parameters</a></li>
</ul></li>
<li class="chapter" data-level="28.6" data-path="population-graphs.html"><a href="population-graphs.html#testing-for-topological-congruence"><i class="fa fa-check"></i><b>28.6</b> Testing for Topological Congruence</a></li>
</ul></li>
<li class="chapter" data-level="29" data-path="interactive-visualization.html"><a href="interactive-visualization.html"><i class="fa fa-check"></i><b>29</b> Interactive Visualization</a><ul>
<li class="chapter" data-level="29.1" data-path="interactive-visualization.html"><a href="interactive-visualization.html#procedure-1"><i class="fa fa-check"></i><b>29.1</b> Procedure</a><ul>
<li class="chapter" data-level="29.1.1" data-path="interactive-visualization.html"><a href="interactive-visualization.html#interactive-networks-1"><i class="fa fa-check"></i><b>29.1.1</b> Interactive Networks</a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>VI Spatial Data</b></span></li>
<li class="chapter" data-level="30" data-path="vector-data.html"><a href="vector-data.html"><i class="fa fa-check"></i><b>30</b> Vector Data</a><ul>
<li class="chapter" data-level="30.1" data-path="vector-data.html"><a href="vector-data.html#points"><i class="fa fa-check"></i><b>30.1</b> Points</a></li>
<li class="chapter" data-level="30.2" data-path="vector-data.html"><a href="vector-data.html#lines"><i class="fa fa-check"></i><b>30.2</b> Lines</a></li>
<li class="chapter" data-level="30.3" data-path="vector-data.html"><a href="vector-data.html#polygons"><i class="fa fa-check"></i><b>30.3</b> Polygons</a></li>
<li class="chapter" data-level="30.4" data-path="vector-data.html"><a href="vector-data.html#saving-vector-objects"><i class="fa fa-check"></i><b>30.4</b> Saving Vector Objects</a></li>
</ul></li>
<li class="chapter" data-level="31" data-path="raster-data.html"><a href="raster-data.html"><i class="fa fa-check"></i><b>31</b> Raster Data</a><ul>
<li class="chapter" data-level="31.1" data-path="raster-data.html"><a href="raster-data.html#cropping-rasters"><i class="fa fa-check"></i><b>31.1</b> Cropping Rasters</a><ul>
<li class="chapter" data-level="31.1.1" data-path="raster-data.html"><a href="raster-data.html#cropping-rasters-via-polygons"><i class="fa fa-check"></i><b>31.1.1</b> Cropping Rasters Via Polygons</a></li>
<li class="chapter" data-level="31.1.2" data-path="raster-data.html"><a href="raster-data.html#cropping-rasters-via-convex-hull"><i class="fa fa-check"></i><b>31.1.2</b> Cropping Rasters Via Convex Hull</a></li>
</ul></li>
<li class="chapter" data-level="31.2" data-path="raster-data.html"><a href="raster-data.html#modifying-rasters"><i class="fa fa-check"></i><b>31.2</b> Modifying Rasters</a></li>
<li class="chapter" data-level="31.3" data-path="raster-data.html"><a href="raster-data.html#extracting-point-data-from-rasters"><i class="fa fa-check"></i><b>31.3</b> Extracting Point Data From Rasters</a><ul>
<li class="chapter" data-level="31.3.1" data-path="raster-data.html"><a href="raster-data.html#stacks-of-rasters"><i class="fa fa-check"></i><b>31.3.1</b> Stacks of Rasters</a></li>
</ul></li>
<li class="chapter" data-level="31.4" data-path="raster-data.html"><a href="raster-data.html#rasters-ggplot"><i class="fa fa-check"></i><b>31.4</b> Rasters &amp; ggplot</a></li>
<li class="chapter" data-level="31.5" data-path="raster-data.html"><a href="raster-data.html#d-visualization"><i class="fa fa-check"></i><b>31.5</b> 3D Visualization</a></li>
<li class="chapter" data-level="31.6" data-path="raster-data.html"><a href="raster-data.html#saving-exporting-rasters"><i class="fa fa-check"></i><b>31.6</b> Saving &amp; Exporting Rasters</a></li>
</ul></li>
<li class="chapter" data-level="32" data-path="map-projections.html"><a href="map-projections.html"><i class="fa fa-check"></i><b>32</b> Map Projections</a><ul>
<li class="chapter" data-level="32.1" data-path="map-projections.html"><a href="map-projections.html#projections"><i class="fa fa-check"></i><b>32.1</b> Projections</a><ul>
<li class="chapter" data-level="32.1.1" data-path="map-projections.html"><a href="map-projections.html#equatorial-projections"><i class="fa fa-check"></i><b>32.1.1</b> Equatorial Projections</a></li>
<li class="chapter" data-level="32.1.2" data-path="map-projections.html"><a href="map-projections.html#azimuth-projections"><i class="fa fa-check"></i><b>32.1.2</b> Azimuth Projections</a></li>
<li class="chapter" data-level="32.1.3" data-path="map-projections.html"><a href="map-projections.html#polar-conic-projections"><i class="fa fa-check"></i><b>32.1.3</b> Polar Conic Projections</a></li>
<li class="chapter" data-level="32.1.4" data-path="map-projections.html"><a href="map-projections.html#miscellaneous-projections"><i class="fa fa-check"></i><b>32.1.4</b> Miscellaneous Projections</a></li>
<li class="chapter" data-level="32.1.5" data-path="map-projections.html"><a href="map-projections.html#reprojecting-rasters"><i class="fa fa-check"></i><b>32.1.5</b> Reprojecting Rasters</a></li>
<li class="chapter" data-level="32.1.6" data-path="map-projections.html"><a href="map-projections.html#projecting-in-ggplot"><i class="fa fa-check"></i><b>32.1.6</b> Projecting In GGPlot</a></li>
</ul></li>
<li class="chapter" data-level="32.2" data-path="map-projections.html"><a href="map-projections.html#coordinate-systems"><i class="fa fa-check"></i><b>32.2</b> Coordinate Systems</a><ul>
<li class="chapter" data-level="32.2.1" data-path="map-projections.html"><a href="map-projections.html#changing-datum"><i class="fa fa-check"></i><b>32.2.1</b> Changing Datum</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="33" data-path="ecological-distance.html"><a href="ecological-distance.html"><i class="fa fa-check"></i><b>33</b> Ecological Distance</a><ul>
<li class="chapter" data-level="33.1" data-path="ecological-distance.html"><a href="ecological-distance.html#the-concept-of-resistance"><i class="fa fa-check"></i><b>33.1</b> The Concept of Resistance</a><ul>
<li class="chapter" data-level="33.1.1" data-path="ecological-distance.html"><a href="ecological-distance.html#absolute-resistance"><i class="fa fa-check"></i><b>33.1.1</b> Absolute Resistance</a></li>
</ul></li>
<li class="chapter" data-level="33.2" data-path="ecological-distance.html"><a href="ecological-distance.html#relative-resistance"><i class="fa fa-check"></i><b>33.2</b> Relative Resistance</a></li>
<li class="chapter" data-level="33.3" data-path="ecological-distance.html"><a href="ecological-distance.html#estimating-separation"><i class="fa fa-check"></i><b>33.3</b> Estimating Separation</a><ul>
<li class="chapter" data-level="33.3.1" data-path="ecological-distance.html"><a href="ecological-distance.html#least-cost-distance"><i class="fa fa-check"></i><b>33.3.1</b> Least Cost Distance</a></li>
<li class="chapter" data-level="33.3.2" data-path="ecological-distance.html"><a href="ecological-distance.html#all-paths-circuit-distance"><i class="fa fa-check"></i><b>33.3.2</b> All Paths (Circuit) Distance</a></li>
<li class="chapter" data-level="33.3.3" data-path="ecological-distance.html"><a href="ecological-distance.html#path-corridor-constraints"><i class="fa fa-check"></i><b>33.3.3</b> Path &amp; Corridor Constraints</a></li>
</ul></li>
<li class="chapter" data-level="33.4" data-path="ecological-distance.html"><a href="ecological-distance.html#estimating-correlations-in-distance"><i class="fa fa-check"></i><b>33.4</b> Estimating Correlations in Distance</a><ul>
<li class="chapter" data-level="33.4.1" data-path="ecological-distance.html"><a href="ecological-distance.html#pairwise-least-cost-path"><i class="fa fa-check"></i><b>33.4.1</b> Pairwise Least Cost Path</a></li>
<li class="chapter" data-level="33.4.2" data-path="ecological-distance.html"><a href="ecological-distance.html#pairwise-all-pairs"><i class="fa fa-check"></i><b>33.4.2</b> Pairwise All Pairs</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="34" data-path="niche-modeling.html"><a href="niche-modeling.html"><i class="fa fa-check"></i><b>34</b> Niche Modeling</a></li>
<li class="chapter" data-level="35" data-path="eigenvector-maps.html"><a href="eigenvector-maps.html"><i class="fa fa-check"></i><b>35</b> Eigenvector Maps</a><ul>
<li class="chapter" data-level="35.1" data-path="eigenvector-maps.html"><a href="eigenvector-maps.html#eigenvectors-eigenvalues"><i class="fa fa-check"></i><b>35.1</b> Eigenvectors &amp; Eigenvalues</a></li>
<li class="chapter" data-level="35.2" data-path="eigenvector-maps.html"><a href="eigenvector-maps.html#principal-coordinates-of-neighborhood-matrices."><i class="fa fa-check"></i><b>35.2</b> Principal Coordinates of Neighborhood Matrices.</a></li>
</ul></li>
<li class="part"><span><b>VII Selection</b></span></li>
<li class="chapter" data-level="36" data-path="classical-selection-models.html"><a href="classical-selection-models.html"><i class="fa fa-check"></i><b>36</b> Classical Selection Models</a><ul>
<li class="chapter" data-level="36.1" data-path="classical-selection-models.html"><a href="classical-selection-models.html#the-strength-of-selection"><i class="fa fa-check"></i><b>36.1</b> The Strength of Selection</a></li>
</ul></li>
<li class="part"><span><b>VIII Backmatter</b></span></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="chapter" data-level="37" data-path="included-data-sets.html"><a href="included-data-sets.html"><i class="fa fa-check"></i><b>37</b> Included Data Sets</a><ul>
<li class="chapter" data-level="37.1" data-path="included-data-sets.html"><a href="included-data-sets.html#the-sonoran-desert-bark-beetle-araptus-attenuata"><i class="fa fa-check"></i><b>37.1</b> The Sonoran desert bark beetle, <em>Araptus attenuata</em></a><ul>
<li class="chapter" data-level="37.1.1" data-path="included-data-sets.html"><a href="included-data-sets.html#nuclear-markers"><i class="fa fa-check"></i><b>37.1.1</b> Nuclear Markers</a></li>
<li class="chapter" data-level="37.1.2" data-path="included-data-sets.html"><a href="included-data-sets.html#methylation-markers"><i class="fa fa-check"></i><b>37.1.2</b> Methylation Markers</a></li>
</ul></li>
<li class="chapter" data-level="37.2" data-path="included-data-sets.html"><a href="included-data-sets.html#the-flowering-dogwood-cornus-florida"><i class="fa fa-check"></i><b>37.2</b> The Flowering Dogwood <em>Cornus florida</em></a></li>
</ul></li>
<li class="chapter" data-level="" data-path="about-the-author.html"><a href="about-the-author.html"><i class="fa fa-check"></i>About the Author</a></li>
<li class="divider"></li>
<li><a href="http://dyerlab.bio.vcu.edu" target="blank">the Dyer Laboratory</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Applied Population Genetics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="mapping-populations" class="section level1 imageChapter">
<h1><span class="header-section-number">16</span> Mapping Populations</h1>
<div class="chapter_image">
<img src="chapter_images/ch_lake.jpg">
</div>
<p>Almost every data set we will work on has a spatial context. Until relatively recently, this was largely ignored and at most given as a 2-dimensional projection of sampling locations on a black and white map in our manuscripts. Here we explore methodologies to provide more intuitive approaches for plotting the spatial location of our sampling sites.</p>
<div id="procedure" class="section level2">
<h2><span class="header-section-number">16.1</span> Procedure</h2>
<p>First, we need to load in the proper libraries. Each of the libraries are needed for:</p>
<ul>
<li><strong>gstudio</strong> - Contains the <em>A. attenuatus</em> data set and there are built-in routines that allow you to plot populations in various formats or export the data in a format for plotting in other systems (e.g., GoogleEarth, ArcGIS, etc).</li>
<li><strong>ggmap</strong> - Allows the plotting of tiles from a googleMap.</li>
<li><strong>leaflet</strong> - Loads in the javascript leaflet library and allows the interactive visualization.</li>
</ul>
</div>
<div id="plotting-population-locations" class="section level2">
<h2><span class="header-section-number">16.2</span> Plotting Population Locations</h2>
<p>We will start by loading in the default data set for <code>gstudio</code> and using methods in that package for producing spatial plots.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># load in the libraries and grab the data</span>
<span class="kw">library</span>(gstudio)
<span class="kw">library</span>(ggmap)
<span class="kw">data</span>(arapat)

<span class="co"># pull out the coordinates - Population is the default stratums</span>
coords &lt;-<span class="st"> </span><span class="kw">strata_coordinates</span>(arapat)
coords$Stratum &lt;-<span class="st"> </span><span class="kw">as.character</span>( coords$Stratum )
coords[<span class="dv">1</span>:<span class="dv">5</span>,]</code></pre></div>
<pre><code>##    Stratum Longitude Latitude
## 1       88 -114.2935 29.32541
## 11       9 -113.9449 29.01457
## 20      84 -113.6679 28.96651
## 29     175 -113.4897 28.72796
## 36     177 -113.9914 28.66056</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># make a map (grabs from google) and plot it</span>
map &lt;-<span class="st"> </span><span class="kw">population_map</span>(coords)
<span class="kw">ggmap</span>( map ) +<span class="st"> </span><span class="kw">geom_point</span>( <span class="kw">aes</span>(<span class="dt">x=</span>Longitude,<span class="dt">y=</span>Latitude),<span class="dt">data=</span>coords)</code></pre></div>
<p><img src="amg_files/figure-html/unnamed-chunk-309-1.png" width="672" /></p>
<p>We can extend this basic plotting by adding text to the plots indicating the population names (here I offset the coordinates by 0.1 &amp; 0.12 degrees—a cheap trick but it works pretty well here).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">p &lt;-<span class="st"> </span><span class="kw">ggmap</span>( map ) +<span class="st"> </span><span class="kw">geom_point</span>( <span class="kw">aes</span>(<span class="dt">x=</span>Longitude,<span class="dt">y=</span>Latitude),<span class="dt">data=</span>coords) 
p +<span class="st"> </span><span class="kw">geom_text</span>( <span class="kw">aes</span>(<span class="dt">x=</span>Longitude<span class="fl">+0.1</span>,<span class="dt">y=</span>Latitude<span class="fl">+0.12</span>,<span class="dt">label=</span>Stratum), <span class="dt">data=</span>coords)</code></pre></div>
<p><img src="amg_files/figure-html/unnamed-chunk-310-1.png" width="672" /></p>
</div>
<div id="population-allele-frequencies" class="section level2">
<h2><span class="header-section-number">16.3</span> Population Allele Frequencies</h2>
<p>It is also possible to plot the data in a spatial context. Here is an example of how to mix <code>ggplot()</code> and <code>ggmap()</code> data and I’ll plot the locations as proportional in size to the allele frequency.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Break out allele frequencies for LTRS by population </span>
f &lt;-<span class="st"> </span><span class="kw">frequencies</span>(arapat,<span class="dt">loci=</span><span class="st">&quot;LTRS&quot;</span>,<span class="dt">stratum=</span><span class="st">&quot;Population&quot;</span>)

<span class="co"># lets only use the allele 01 </span>
f &lt;-<span class="st"> </span>f[ f$Allele==<span class="st">&quot;01&quot;</span>,]

<span class="co"># here are what the data look like</span>
f[<span class="dv">1</span>:<span class="dv">10</span>,]</code></pre></div>
<pre><code>##    Stratum Locus Allele Frequency
## 1      101  LTRS     01 0.2777778
## 3      102  LTRS     01 0.3125000
## 5       12  LTRS     01 0.8000000
## 7      153  LTRS     01 0.1500000
## 11     159  LTRS     01 0.8888889
## 13     160  LTRS     01 0.8000000
## 15     161  LTRS     01 0.9500000
## 17     162  LTRS     01 0.9500000
## 19     163  LTRS     01 0.1000000
## 21     164  LTRS     01 0.3500000</code></pre>
<p>and then use those frequencies as image size.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># merget the frequencies into the coordinate data.frame</span>
coords &lt;-<span class="st"> </span><span class="kw">merge</span>( coords, f[,<span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">4</span>)])
<span class="kw">ggmap</span>( map ) +<span class="st"> </span><span class="kw">geom_point</span>( <span class="kw">aes</span>(<span class="dt">x=</span>Longitude,<span class="dt">y=</span>Latitude,<span class="dt">size=</span>Frequency),<span class="dt">data=</span>coords)</code></pre></div>
<p><img src="amg_files/figure-html/unnamed-chunk-312-1.png" width="672" /></p>
<p>or for observed effective allele size</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># grab effect for LTRS locus</span>
ae &lt;-<span class="st"> </span><span class="kw">genetic_diversity</span>(arapat, <span class="dt">stratum =</span> <span class="st">&quot;Population&quot;</span>, <span class="dt">mode=</span><span class="st">&quot;Ae&quot;</span>)
ae &lt;-<span class="st"> </span>ae[ ae$Locus==<span class="st">&quot;LTRS&quot;</span>,]

<span class="co"># merge in the data</span>
coords &lt;-<span class="st"> </span><span class="kw">merge</span>( coords, ae[,<span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">3</span>)])
<span class="kw">ggmap</span>( map ) +<span class="st"> </span><span class="kw">geom_point</span>( <span class="kw">aes</span>(<span class="dt">x=</span>Longitude,<span class="dt">y=</span>Latitude,<span class="dt">size=</span>Ae),<span class="dt">data=</span>coords)</code></pre></div>
<p><img src="amg_files/figure-html/unnamed-chunk-313-1.png" width="672" /></p>
<p>There is also the option to make use of some pie charts. I know, pie charts suck and any statistician will tell you that they should probably not be used because they can be misleading, but here they are. For exploratory data analysis, they can be very insightful at times. Here is the frequency of alleles at the <em>enolase</em> locus in <em>Araptus</em>. Any spatial structuring catch your eye?</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">pies_on_map</span>( arapat, <span class="dt">stratum=</span><span class="st">&quot;Population&quot;</span>, <span class="dt">locus=</span><span class="st">&quot;EN&quot;</span>)</code></pre></div>
<p>Which will open a new browser window and produce a graph like the one below.</p>
<iframe width="640" height="480" src="media/pies_on_map.html" frameborder="0" allowfullscreen>
</iframe>
<p>Note the messages about the approximation. This is because the google maps API has an integer for zoom factor and at times it is not able to get all the points into the field of view using an integer zoom. If this happens to you, you can manually specify the <em>zoom</em> as an optional argument to either function <code>pies_on_map()</code> or <code>population_map()</code>. You also need to be careful with the <code>pies_on_map()</code> function because the way it works is that the background tile is plotted and then I plot the pies ontop of it. If you reshape your plot window outside equal x- and y- coordinates (e.g., make it a non-square figure), the spatial location of the pie charts will move! This is a very frustrating thing but it has to do with the way viewports are overlain onto graphical objects in R and I have absolutely <strong>no</strong> control over it. So, the best option is to make the plot square, export it as PNG or TIFF or whaterver, then crop as necessary.</p>
</div>
<div id="dynamic-maps" class="section level2">
<h2><span class="header-section-number">16.4</span> Dynamic maps</h2>
<p>We can also produce dynamic maps for viewing in html output (like ebooks or webpages). Here I produce a leaflet map and then add markers on to the file using coordinates and put the clicked popup as the population name. This is an <em>interactive</em> map, you can zoom in, etc.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(leaflet)
map &lt;-<span class="st"> </span><span class="kw">addTiles</span>( <span class="kw">leaflet</span>( <span class="dt">data=</span>coords) )
map &lt;-<span class="st"> </span><span class="kw">addMarkers</span>( ~Longitude, ~Latitude, <span class="dt">popup=</span>~Stratum , <span class="dt">map=</span>map)
map</code></pre></div>
<div id="htmlwidget-a9e1c6955d9b93264920" style="width:672px;height:480px;" class="leaflet html-widget"></div>
<script type="application/json" data-for="htmlwidget-a9e1c6955d9b93264920">{"x":{"options":{"crs":{"crsClass":"L.CRS.EPSG3857","code":null,"proj4def":null,"projectedBounds":null,"options":{}}},"calls":[{"method":"addTiles","args":["//{s}.tile.openstreetmap.org/{z}/{x}/{y}.png",null,null,{"minZoom":0,"maxZoom":18,"maxNativeZoom":null,"tileSize":256,"subdomains":"abc","errorTileUrl":"","tms":false,"continuousWorld":false,"noWrap":false,"zoomOffset":0,"zoomReverse":false,"opacity":1,"zIndex":null,"unloadInvisibleTiles":null,"updateWhenIdle":null,"detectRetina":false,"reuseTiles":false,"attribution":"&copy; <a href=\"http://openstreetmap.org\">OpenStreetMap<\/a> contributors, <a href=\"http://creativecommons.org/licenses/by-sa/2.0/\">CC-BY-SA<\/a>"}]},{"method":"addMarkers","args":[[27.90509,26.38014,27.18232,24.13389,27.52944,27.40498,27.0367,27.2028,24.2115,24.74642,26.24905,25.91409,25.55757,26.20876,28.22308,28.40846,28.72796,28.66056,26.63783,25.34819,26.0155,25.60521,24.58843,24.87611,28.96651,29.32541,28.03661,29.01457,26.94589,23.2855,25.0247,23.08984,27.3632],[-110.57436,-109.12633,-112.66552,-110.46236,-113.31609,-112.52959,-112.986,-112.408,-110.951,-111.5441,-112.40948,-112.08062,-111.21563,-111.37833,-113.18263,-112.86985,-113.48973,-113.99141,-109.327,-111.60056,-111.35475,-111.32638,-110.74599,-110.69175,-113.66787,-114.29353,-113.39991,-113.94486,-112.04613,-110.10429,-111.675,-110.1091,-112.964],null,null,null,{"clickable":true,"draggable":false,"keyboard":true,"title":"","alt":"","zIndexOffset":0,"opacity":1,"riseOnHover":false,"riseOffset":250},["101","102","12","153","159","160","161","162","163","164","165","166","168","169","171","173","175","177","32","51","58","64","75","77","84","88","89","9","93","Aqu","Const","Mat","SFr"],null,null,null,null,null,null]}],"limits":{"lat":[23.08984,29.32541],"lng":[-114.29353,-109.12633]}},"evals":[],"jsHooks":[]}</script>
<p>You notice the interesting <code>%&gt;%</code> operator? This is short-hand that may be very helpful for you. It is part of the <code>magrittr</code> library, and is often imported through <code>dplyr</code> and not directly. However, what it does is allow us to be a bit more succinct that does essentially the same thing as</p>
<p>You can change the underlying base map by passing on other Tiles to the construction. A list of available tile providers can be found <a href="http://leaflet-extras.github.io/leaflet-providers/preview/index.html">here</a>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">map &lt;-<span class="st"> </span><span class="kw">leaflet</span>( <span class="dt">data=</span>coords )
map &lt;-<span class="st"> </span><span class="kw">addProviderTiles</span>(<span class="st">&quot;MapQuestOpen.Aerial&quot;</span>, <span class="dt">map=</span>map)
<span class="kw">addMarkers</span>( ~Longitude, ~Latitude, <span class="dt">popup=</span>~Stratum, <span class="dt">map=</span>map)</code></pre></div>
<div id="htmlwidget-f6a84d53bb2653cdddca" style="width:672px;height:480px;" class="leaflet html-widget"></div>
<script type="application/json" data-for="htmlwidget-f6a84d53bb2653cdddca">{"x":{"options":{"crs":{"crsClass":"L.CRS.EPSG3857","code":null,"proj4def":null,"projectedBounds":null,"options":{}}},"calls":[{"method":"addProviderTiles","args":["MapQuestOpen.Aerial",null,null,{"errorTileUrl":"","noWrap":false,"zIndex":null,"unloadInvisibleTiles":null,"updateWhenIdle":null,"detectRetina":false,"reuseTiles":false}]},{"method":"addMarkers","args":[[27.90509,26.38014,27.18232,24.13389,27.52944,27.40498,27.0367,27.2028,24.2115,24.74642,26.24905,25.91409,25.55757,26.20876,28.22308,28.40846,28.72796,28.66056,26.63783,25.34819,26.0155,25.60521,24.58843,24.87611,28.96651,29.32541,28.03661,29.01457,26.94589,23.2855,25.0247,23.08984,27.3632],[-110.57436,-109.12633,-112.66552,-110.46236,-113.31609,-112.52959,-112.986,-112.408,-110.951,-111.5441,-112.40948,-112.08062,-111.21563,-111.37833,-113.18263,-112.86985,-113.48973,-113.99141,-109.327,-111.60056,-111.35475,-111.32638,-110.74599,-110.69175,-113.66787,-114.29353,-113.39991,-113.94486,-112.04613,-110.10429,-111.675,-110.1091,-112.964],null,null,null,{"clickable":true,"draggable":false,"keyboard":true,"title":"","alt":"","zIndexOffset":0,"opacity":1,"riseOnHover":false,"riseOffset":250},["101","102","12","153","159","160","161","162","163","164","165","166","168","169","171","173","175","177","32","51","58","64","75","77","84","88","89","9","93","Aqu","Const","Mat","SFr"],null,null,null,null,null,null]}],"limits":{"lat":[23.08984,29.32541],"lng":[-114.29353,-109.12633]}},"evals":[],"jsHooks":[]}</script>
<p>Or you can add custom images for the map relative to some meaningful information.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">coords$Size &lt;-<span class="st"> </span><span class="kw">log</span>(coords$Ae) *<span class="st"> </span><span class="dv">10</span> +<span class="st"> </span><span class="dv">1</span>
<span class="kw">addCircleMarkers</span>( ~Longitude, ~Latitude, <span class="dt">popup=</span>~Stratum, <span class="dt">radius=</span>~Size, <span class="dt">color=</span><span class="st">&quot;red&quot;</span>, <span class="dt">fillOpacity=</span><span class="fl">0.75</span>, <span class="dt">data =</span> coords, <span class="dt">map =</span> map) </code></pre></div>
<div id="htmlwidget-0591b459c543257390b5" style="width:672px;height:480px;" class="leaflet html-widget"></div>
<script type="application/json" data-for="htmlwidget-0591b459c543257390b5">{"x":{"options":{"crs":{"crsClass":"L.CRS.EPSG3857","code":null,"proj4def":null,"projectedBounds":null,"options":{}}},"calls":[{"method":"addProviderTiles","args":["MapQuestOpen.Aerial",null,null,{"errorTileUrl":"","noWrap":false,"zIndex":null,"unloadInvisibleTiles":null,"updateWhenIdle":null,"detectRetina":false,"reuseTiles":false}]},{"method":"addCircleMarkers","args":[[27.90509,26.38014,27.18232,24.13389,27.52944,27.40498,27.0367,27.2028,24.2115,24.74642,26.24905,25.91409,25.55757,26.20876,28.22308,28.40846,28.72796,28.66056,26.63783,25.34819,26.0155,25.60521,24.58843,24.87611,28.96651,29.32541,28.03661,29.01457,26.94589,23.2855,25.0247,23.08984,27.3632],[-110.57436,-109.12633,-112.66552,-110.46236,-113.31609,-112.52959,-112.986,-112.408,-110.951,-111.5441,-112.40948,-112.08062,-111.21563,-111.37833,-113.18263,-112.86985,-113.48973,-113.99141,-109.327,-111.60056,-111.35475,-111.32638,-110.74599,-110.69175,-113.66787,-114.29353,-113.39991,-113.94486,-112.04613,-110.10429,-111.675,-110.1091,-112.964],[6.12885356729001,6.61570822771226,4.85662480811984,3.94371060602578,3.20061884776802,4.85662480811984,1.99820335282211,1.99820335282211,2.98450938723838,7.06969484318893,6.44727175441672,6.44727175441672,5.70003629245736,7.06969484318893,2.98450938723838,2.98450938723838,1,1,5.48627095896178,3.80902385466402,5.24157241120317,6.44727175441672,2.80670028755482,7.53926467406664,1,1,2.98450938723838,1,6.44727175441672,4.63393894187477,4.53139289106424,2.98450938723838,5.24157241120317],null,null,{"lineCap":null,"lineJoin":null,"clickable":true,"pointerEvents":null,"className":"","stroke":true,"color":"red","weight":5,"opacity":0.5,"fill":true,"fillColor":"red","fillOpacity":0.75,"dashArray":null},null,null,["101","102","12","153","159","160","161","162","163","164","165","166","168","169","171","173","175","177","32","51","58","64","75","77","84","88","89","9","93","Aqu","Const","Mat","SFr"],null,null,null,null]}],"limits":{"lat":[23.08984,29.32541],"lng":[-114.29353,-109.12633]}},"evals":[],"jsHooks":[]}</script>
<p>In the information that pops out of the marker, you can put a bit more information that may be helpful. Here I estimate some diversity measures (H<sub>O</sub>, H<sub>E</sub>, and A<sub>E</sub>).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">x &lt;-<span class="st"> </span><span class="kw">genetic_diversity</span>(arapat,<span class="dt">stratum =</span> <span class="st">&quot;Population&quot;</span>,<span class="dt">mode=</span><span class="st">&quot;Ho&quot;</span>)
x &lt;-<span class="st"> </span>x[ x$Locus ==<span class="st"> &quot;LTRS&quot;</span>,]
coords &lt;-<span class="st"> </span><span class="kw">merge</span>( coords, x[,<span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">3</span>)])
x &lt;-<span class="st"> </span><span class="kw">genetic_diversity</span>(arapat,<span class="dt">stratum =</span> <span class="st">&quot;Population&quot;</span>,<span class="dt">mode=</span><span class="st">&quot;He&quot;</span>)
x &lt;-<span class="st"> </span>x[ x$Locus ==<span class="st"> &quot;LTRS&quot;</span>,]
coords &lt;-<span class="st"> </span><span class="kw">merge</span>( coords, x[,<span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">3</span>)])
x &lt;-<span class="st"> </span><span class="kw">genetic_diversity</span>(arapat,<span class="dt">stratum =</span> <span class="st">&quot;Population&quot;</span>,<span class="dt">mode=</span><span class="st">&quot;Ae&quot;</span>)
x &lt;-<span class="st"> </span>x[ x$Locus ==<span class="st"> &quot;LTRS&quot;</span>,]
coords &lt;-<span class="st"> </span><span class="kw">merge</span>( coords, x[,<span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">3</span>)])</code></pre></div>
<p>Make these data into a table format in html (OK it is a bit ugly but it is reasonable.)</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">coords$label &lt;-<span class="st"> </span><span class="kw">paste</span>(<span class="st">&quot;&lt;h2&gt;&quot;</span>,coords$Stratum,<span class="st">&quot;&lt;/h2&gt;&lt;table&gt;&lt;tr&gt;&lt;th&gt;Parameter&lt;/th&gt;&lt;th&gt;Value&lt;/th&gt;&lt;/tr&gt;&lt;tr&gt;&lt;td&gt;H&lt;sub&gt;O&lt;/sub&gt;&lt;/td&gt;&lt;td&gt;&quot;</span>, <span class="kw">format</span>(coords$Ho,<span class="dt">digits =</span> <span class="dv">3</span>),<span class="st">&quot;&lt;/td&gt;&lt;/tr&gt;&lt;tr&gt;&lt;td&gt;H&lt;sub&gt;E&lt;/sub&gt;&lt;/td&gt;&lt;td&gt;&quot;</span>,<span class="kw">format</span>(coords$He, <span class="dt">digits=</span><span class="dv">3</span>), <span class="st">&quot;&lt;/td&gt;&lt;/tr&gt;&lt;tr&gt;&lt;td&gt;A&lt;sub&gt;E&lt;/sub&gt;&lt;/td&gt;&lt;td&gt;&quot;</span>,<span class="kw">format</span>(coords$Ae,<span class="dt">digits=</span><span class="dv">3</span>),<span class="st">&quot;&lt;/td&gt;&lt;/tr&gt;&lt;/table&gt;&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;&quot;</span>)

<span class="co"># plot it</span>
map &lt;-<span class="st"> </span><span class="kw">addTiles</span>( <span class="kw">leaflet</span>( <span class="dt">data=</span>coords) )
<span class="kw">addCircleMarkers</span>( ~Longitude, ~Latitude, <span class="dt">popup=</span>~label, <span class="dt">radius=</span>~Size, <span class="dt">color=</span><span class="st">&quot;red&quot;</span>, <span class="dt">fillOpacity=</span><span class="fl">0.75</span>, <span class="dt">map =</span> map, <span class="dt">data =</span> coords) </code></pre></div>
<div id="htmlwidget-e3d154bbe185c6620605" style="width:672px;height:480px;" class="leaflet html-widget"></div>
<script type="application/json" data-for="htmlwidget-e3d154bbe185c6620605">{"x":{"options":{"crs":{"crsClass":"L.CRS.EPSG3857","code":null,"proj4def":null,"projectedBounds":null,"options":{}}},"calls":[{"method":"addTiles","args":["//{s}.tile.openstreetmap.org/{z}/{x}/{y}.png",null,null,{"minZoom":0,"maxZoom":18,"maxNativeZoom":null,"tileSize":256,"subdomains":"abc","errorTileUrl":"","tms":false,"continuousWorld":false,"noWrap":false,"zoomOffset":0,"zoomReverse":false,"opacity":1,"zIndex":null,"unloadInvisibleTiles":null,"updateWhenIdle":null,"detectRetina":false,"reuseTiles":false,"attribution":"&copy; <a href=\"http://openstreetmap.org\">OpenStreetMap<\/a> contributors, <a href=\"http://creativecommons.org/licenses/by-sa/2.0/\">CC-BY-SA<\/a>"}]},{"method":"addCircleMarkers","args":[[27.90509,26.38014,27.18232,24.13389,27.52944,27.40498,27.0367,27.2028,24.2115,24.74642,26.24905,25.91409,25.55757,26.20876,28.22308,28.40846,28.72796,28.66056,26.63783,25.34819,26.0155,25.60521,24.58843,24.87611,28.96651,29.32541,28.03661,29.01457,26.94589,23.2855,25.0247,23.08984,27.3632],[-110.57436,-109.12633,-112.66552,-110.46236,-113.31609,-112.52959,-112.986,-112.408,-110.951,-111.5441,-112.40948,-112.08062,-111.21563,-111.37833,-113.18263,-112.86985,-113.48973,-113.99141,-109.327,-111.60056,-111.35475,-111.32638,-110.74599,-110.69175,-113.66787,-114.29353,-113.39991,-113.94486,-112.04613,-110.10429,-111.675,-110.1091,-112.964],[6.12885356729001,6.61570822771226,4.85662480811984,3.94371060602578,3.20061884776802,4.85662480811984,1.99820335282211,1.99820335282211,2.98450938723838,7.06969484318893,6.44727175441672,6.44727175441672,5.70003629245736,7.06969484318893,2.98450938723838,2.98450938723838,1,1,5.48627095896178,3.80902385466402,5.24157241120317,6.44727175441672,2.80670028755482,7.53926467406664,1,1,2.98450938723838,1,6.44727175441672,4.63393894187477,4.53139289106424,2.98450938723838,5.24157241120317],null,null,{"lineCap":null,"lineJoin":null,"clickable":true,"pointerEvents":null,"className":"","stroke":true,"color":"red","weight":5,"opacity":0.5,"fill":true,"fillColor":"red","fillOpacity":0.75,"dashArray":null},null,null,["<h2>101<\/h2><table><tr><th>Parameter<\/th><th>Value<\/th><\/tr><tr><td>H<sub>O<\/sub><\/td><td>0.556<\/td><\/tr><tr><td>H<sub>E<\/sub><\/td><td>0.401<\/td><\/tr><tr><td>A<sub>E<\/sub><\/td><td>1.67<\/td><\/tr><\/table>","<h2>102<\/h2><table><tr><th>Parameter<\/th><th>Value<\/th><\/tr><tr><td>H<sub>O<\/sub><\/td><td>0.625<\/td><\/tr><tr><td>H<sub>E<\/sub><\/td><td>0.430<\/td><\/tr><tr><td>A<sub>E<\/sub><\/td><td>1.75<\/td><\/tr><\/table>","<h2>12<\/h2><table><tr><th>Parameter<\/th><th>Value<\/th><\/tr><tr><td>H<sub>O<\/sub><\/td><td>0.400<\/td><\/tr><tr><td>H<sub>E<\/sub><\/td><td>0.320<\/td><\/tr><tr><td>A<sub>E<\/sub><\/td><td>1.47<\/td><\/tr><\/table>","<h2>153<\/h2><table><tr><th>Parameter<\/th><th>Value<\/th><\/tr><tr><td>H<sub>O<\/sub><\/td><td>0.300<\/td><\/tr><tr><td>H<sub>E<\/sub><\/td><td>0.255<\/td><\/tr><tr><td>A<sub>E<\/sub><\/td><td>1.34<\/td><\/tr><\/table>","<h2>159<\/h2><table><tr><th>Parameter<\/th><th>Value<\/th><\/tr><tr><td>H<sub>O<\/sub><\/td><td>0.222<\/td><\/tr><tr><td>H<sub>E<\/sub><\/td><td>0.198<\/td><\/tr><tr><td>A<sub>E<\/sub><\/td><td>1.25<\/td><\/tr><\/table>","<h2>160<\/h2><table><tr><th>Parameter<\/th><th>Value<\/th><\/tr><tr><td>H<sub>O<\/sub><\/td><td>0.200<\/td><\/tr><tr><td>H<sub>E<\/sub><\/td><td>0.320<\/td><\/tr><tr><td>A<sub>E<\/sub><\/td><td>1.47<\/td><\/tr><\/table>","<h2>161<\/h2><table><tr><th>Parameter<\/th><th>Value<\/th><\/tr><tr><td>H<sub>O<\/sub><\/td><td>0.100<\/td><\/tr><tr><td>H<sub>E<\/sub><\/td><td>0.095<\/td><\/tr><tr><td>A<sub>E<\/sub><\/td><td>1.10<\/td><\/tr><\/table>","<h2>162<\/h2><table><tr><th>Parameter<\/th><th>Value<\/th><\/tr><tr><td>H<sub>O<\/sub><\/td><td>0.100<\/td><\/tr><tr><td>H<sub>E<\/sub><\/td><td>0.095<\/td><\/tr><tr><td>A<sub>E<\/sub><\/td><td>1.10<\/td><\/tr><\/table>","<h2>163<\/h2><table><tr><th>Parameter<\/th><th>Value<\/th><\/tr><tr><td>H<sub>O<\/sub><\/td><td>0.200<\/td><\/tr><tr><td>H<sub>E<\/sub><\/td><td>0.180<\/td><\/tr><tr><td>A<sub>E<\/sub><\/td><td>1.22<\/td><\/tr><\/table>","<h2>164<\/h2><table><tr><th>Parameter<\/th><th>Value<\/th><\/tr><tr><td>H<sub>O<\/sub><\/td><td>0.300<\/td><\/tr><tr><td>H<sub>E<\/sub><\/td><td>0.455<\/td><\/tr><tr><td>A<sub>E<\/sub><\/td><td>1.83<\/td><\/tr><\/table>","<h2>165<\/h2><table><tr><th>Parameter<\/th><th>Value<\/th><\/tr><tr><td>H<sub>O<\/sub><\/td><td>0.400<\/td><\/tr><tr><td>H<sub>E<\/sub><\/td><td>0.420<\/td><\/tr><tr><td>A<sub>E<\/sub><\/td><td>1.72<\/td><\/tr><\/table>","<h2>166<\/h2><table><tr><th>Parameter<\/th><th>Value<\/th><\/tr><tr><td>H<sub>O<\/sub><\/td><td>0.200<\/td><\/tr><tr><td>H<sub>E<\/sub><\/td><td>0.420<\/td><\/tr><tr><td>A<sub>E<\/sub><\/td><td>1.72<\/td><\/tr><\/table>","<h2>168<\/h2><table><tr><th>Parameter<\/th><th>Value<\/th><\/tr><tr><td>H<sub>O<\/sub><\/td><td>0.500<\/td><\/tr><tr><td>H<sub>E<\/sub><\/td><td>0.375<\/td><\/tr><tr><td>A<sub>E<\/sub><\/td><td>1.60<\/td><\/tr><\/table>","<h2>169<\/h2><table><tr><th>Parameter<\/th><th>Value<\/th><\/tr><tr><td>H<sub>O<\/sub><\/td><td>0.500<\/td><\/tr><tr><td>H<sub>E<\/sub><\/td><td>0.455<\/td><\/tr><tr><td>A<sub>E<\/sub><\/td><td>1.83<\/td><\/tr><\/table>","<h2>171<\/h2><table><tr><th>Parameter<\/th><th>Value<\/th><\/tr><tr><td>H<sub>O<\/sub><\/td><td>0.200<\/td><\/tr><tr><td>H<sub>E<\/sub><\/td><td>0.180<\/td><\/tr><tr><td>A<sub>E<\/sub><\/td><td>1.22<\/td><\/tr><\/table>","<h2>173<\/h2><table><tr><th>Parameter<\/th><th>Value<\/th><\/tr><tr><td>H<sub>O<\/sub><\/td><td>0.200<\/td><\/tr><tr><td>H<sub>E<\/sub><\/td><td>0.180<\/td><\/tr><tr><td>A<sub>E<\/sub><\/td><td>1.22<\/td><\/tr><\/table>","<h2>175<\/h2><table><tr><th>Parameter<\/th><th>Value<\/th><\/tr><tr><td>H<sub>O<\/sub><\/td><td>0.000<\/td><\/tr><tr><td>H<sub>E<\/sub><\/td><td>0.000<\/td><\/tr><tr><td>A<sub>E<\/sub><\/td><td>1.00<\/td><\/tr><\/table>","<h2>177<\/h2><table><tr><th>Parameter<\/th><th>Value<\/th><\/tr><tr><td>H<sub>O<\/sub><\/td><td>0.000<\/td><\/tr><tr><td>H<sub>E<\/sub><\/td><td>0.000<\/td><\/tr><tr><td>A<sub>E<\/sub><\/td><td>1.00<\/td><\/tr><\/table>","<h2>32<\/h2><table><tr><th>Parameter<\/th><th>Value<\/th><\/tr><tr><td>H<sub>O<\/sub><\/td><td>0.368<\/td><\/tr><tr><td>H<sub>E<\/sub><\/td><td>0.361<\/td><\/tr><tr><td>A<sub>E<\/sub><\/td><td>1.57<\/td><\/tr><\/table>","<h2>51<\/h2><table><tr><th>Parameter<\/th><th>Value<\/th><\/tr><tr><td>H<sub>O<\/sub><\/td><td>0.286<\/td><\/tr><tr><td>H<sub>E<\/sub><\/td><td>0.245<\/td><\/tr><tr><td>A<sub>E<\/sub><\/td><td>1.32<\/td><\/tr><\/table>","<h2>58<\/h2><table><tr><th>Parameter<\/th><th>Value<\/th><\/tr><tr><td>H<sub>O<\/sub><\/td><td>0.444<\/td><\/tr><tr><td>H<sub>E<\/sub><\/td><td>0.346<\/td><\/tr><tr><td>A<sub>E<\/sub><\/td><td>1.53<\/td><\/tr><\/table>","<h2>64<\/h2><table><tr><th>Parameter<\/th><th>Value<\/th><\/tr><tr><td>H<sub>O<\/sub><\/td><td>0.200<\/td><\/tr><tr><td>H<sub>E<\/sub><\/td><td>0.420<\/td><\/tr><tr><td>A<sub>E<\/sub><\/td><td>1.72<\/td><\/tr><\/table>","<h2>75<\/h2><table><tr><th>Parameter<\/th><th>Value<\/th><\/tr><tr><td>H<sub>O<\/sub><\/td><td>0.182<\/td><\/tr><tr><td>H<sub>E<\/sub><\/td><td>0.165<\/td><\/tr><tr><td>A<sub>E<\/sub><\/td><td>1.20<\/td><\/tr><\/table>","<h2>77<\/h2><table><tr><th>Parameter<\/th><th>Value<\/th><\/tr><tr><td>H<sub>O<\/sub><\/td><td>0.400<\/td><\/tr><tr><td>H<sub>E<\/sub><\/td><td>0.480<\/td><\/tr><tr><td>A<sub>E<\/sub><\/td><td>1.92<\/td><\/tr><\/table>","<h2>84<\/h2><table><tr><th>Parameter<\/th><th>Value<\/th><\/tr><tr><td>H<sub>O<\/sub><\/td><td>0.000<\/td><\/tr><tr><td>H<sub>E<\/sub><\/td><td>0.000<\/td><\/tr><tr><td>A<sub>E<\/sub><\/td><td>1.00<\/td><\/tr><\/table>","<h2>88<\/h2><table><tr><th>Parameter<\/th><th>Value<\/th><\/tr><tr><td>H<sub>O<\/sub><\/td><td>0.000<\/td><\/tr><tr><td>H<sub>E<\/sub><\/td><td>0.000<\/td><\/tr><tr><td>A<sub>E<\/sub><\/td><td>1.00<\/td><\/tr><\/table>","<h2>89<\/h2><table><tr><th>Parameter<\/th><th>Value<\/th><\/tr><tr><td>H<sub>O<\/sub><\/td><td>0.200<\/td><\/tr><tr><td>H<sub>E<\/sub><\/td><td>0.180<\/td><\/tr><tr><td>A<sub>E<\/sub><\/td><td>1.22<\/td><\/tr><\/table>","<h2>9<\/h2><table><tr><th>Parameter<\/th><th>Value<\/th><\/tr><tr><td>H<sub>O<\/sub><\/td><td>0.000<\/td><\/tr><tr><td>H<sub>E<\/sub><\/td><td>0.000<\/td><\/tr><tr><td>A<sub>E<\/sub><\/td><td>1.00<\/td><\/tr><\/table>","<h2>93<\/h2><table><tr><th>Parameter<\/th><th>Value<\/th><\/tr><tr><td>H<sub>O<\/sub><\/td><td>0.600<\/td><\/tr><tr><td>H<sub>E<\/sub><\/td><td>0.420<\/td><\/tr><tr><td>A<sub>E<\/sub><\/td><td>1.72<\/td><\/tr><\/table>","<h2>Aqu<\/h2><table><tr><th>Parameter<\/th><th>Value<\/th><\/tr><tr><td>H<sub>O<\/sub><\/td><td>0.375<\/td><\/tr><tr><td>H<sub>E<\/sub><\/td><td>0.305<\/td><\/tr><tr><td>A<sub>E<\/sub><\/td><td>1.44<\/td><\/tr><\/table>","<h2>Const<\/h2><table><tr><th>Parameter<\/th><th>Value<\/th><\/tr><tr><td>H<sub>O<\/sub><\/td><td>0.364<\/td><\/tr><tr><td>H<sub>E<\/sub><\/td><td>0.298<\/td><\/tr><tr><td>A<sub>E<\/sub><\/td><td>1.42<\/td><\/tr><\/table>","<h2>Mat<\/h2><table><tr><th>Parameter<\/th><th>Value<\/th><\/tr><tr><td>H<sub>O<\/sub><\/td><td>0.200<\/td><\/tr><tr><td>H<sub>E<\/sub><\/td><td>0.180<\/td><\/tr><tr><td>A<sub>E<\/sub><\/td><td>1.22<\/td><\/tr><\/table>","<h2>SFr<\/h2><table><tr><th>Parameter<\/th><th>Value<\/th><\/tr><tr><td>H<sub>O<\/sub><\/td><td>0.222<\/td><\/tr><tr><td>H<sub>E<\/sub><\/td><td>0.346<\/td><\/tr><tr><td>A<sub>E<\/sub><\/td><td>1.53<\/td><\/tr><\/table>"],null,null,null,null]}],"limits":{"lat":[23.08984,29.32541],"lng":[-114.29353,-109.12633]}},"evals":[],"jsHooks":[]}</script>
<div id="sankey-networks" class="section level3">
<h3><span class="header-section-number">16.4.1</span> Sankey Networks</h3>
<p>A Sankey Diagram is a kind of flow diagram used to indicate allocation of one set of object into two other sets. This is a complicated set of interactions to display. In the following example, I’ll use the populations from the <em>A. attenuata</em> data set and then map each individual onto its STRUCTURE cluster and mtDNA clade. This plot needs a <code>data.frame</code> for nodes and one for edges created in a particular way.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"> <span class="kw">library</span>(networkD3)
pops &lt;-<span class="st"> </span><span class="kw">as.character</span>(<span class="kw">unique</span>(arapat$Population))
species &lt;-<span class="st"> </span><span class="kw">as.character</span>(<span class="kw">unique</span>(arapat$Species))
cluster &lt;-<span class="st"> </span><span class="kw">as.character</span>(<span class="kw">unique</span>(arapat$Cluster))

sankey.nodes &lt;-<span class="st"> </span><span class="kw">data.frame</span>( <span class="dt">name =</span> <span class="kw">c</span>( pops, species, cluster), 
                            <span class="dt">stringsAsFactors =</span> <span class="ot">FALSE</span>)


<span class="co"># takes vector of values and finds out the number of times</span>
<span class="co"># each combination is found in the pair.</span>
get_sets &lt;-<span class="st"> </span>function( source, target ) {
  t &lt;-<span class="st"> </span><span class="kw">table</span>(source, target)
  m &lt;-<span class="st"> </span><span class="kw">which</span>( t &gt;<span class="st"> </span><span class="dv">0</span>, <span class="dt">arr.ind =</span> <span class="ot">TRUE</span>)
  <span class="kw">suppressWarnings</span>( df &lt;-<span class="st"> </span><span class="kw">data.frame</span>( <span class="dt">ind=</span>m ) )
  df$source &lt;-<span class="st"> </span><span class="kw">rownames</span>(t)[df$ind.source]
  df$target &lt;-<span class="st"> </span><span class="kw">colnames</span>(t)[df$ind.target]
  df$value &lt;-<span class="st"> </span><span class="kw">sapply</span>(<span class="kw">seq</span>(<span class="dv">1</span>,<span class="kw">dim</span>(df)[<span class="dv">1</span>]), 
                     function(x) {
                         <span class="kw">return</span>(t[df$ind.source[x],df$ind.target[x]])
                         })
  <span class="kw">return</span>( df[,<span class="dv">3</span>:<span class="dv">5</span>] )
}

<span class="co"># For finding the number of the name instead of the name</span>
nameIndex &lt;-<span class="st"> </span>function(vals,names){
  ret &lt;-<span class="st"> </span><span class="kw">sapply</span>( vals, function(x) {<span class="kw">which</span>(x==names)-<span class="dv">1</span>}, <span class="dt">USE.NAMES =</span> <span class="ot">FALSE</span>)
  <span class="kw">return</span>(ret)
}



sankey.edges &lt;-<span class="st"> </span><span class="kw">rbind</span>( <span class="kw">get_sets</span>(arapat$Species,arapat$Population),
                       <span class="kw">get_sets</span>(arapat$Population, arapat$Cluster) )
sankey.edges$targetNum &lt;-<span class="st"> </span><span class="kw">nameIndex</span>(sankey.edges$target, sankey.nodes$name)
sankey.edges$sourceNum &lt;-<span class="st"> </span><span class="kw">nameIndex</span>(sankey.edges$source, sankey.nodes$name)</code></pre></div>
<p>What it does produce is a very cool and dynamic plot. The colored boxes represeent the groups–Species &amp; STRUCTURE Cluster on the outside, population on the inside. Connecting lines indicate individuals who are in each of these partially overlapping groups. For example, if you grab the population <em>Mat</em>, you will see it has samples in it who are identified as either mtDNA type <em>Peninsual</em> or <em>Cape</em> as well as belonging to STRUCTURE groups <em>CBP-C</em> and <em>SCBP-A</em>. If you have difficult data such as these, this kind of plot may be very helpful.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">sankeyNetwork</span>( <span class="dt">Links=</span>sankey.edges, <span class="dt">Nodes=</span>sankey.nodes,
               <span class="dt">Source =</span> <span class="st">&quot;sourceNum&quot;</span>, <span class="dt">Target =</span> <span class="st">&quot;targetNum&quot;</span>,
               <span class="dt">Value =</span> <span class="st">&quot;value&quot;</span>, <span class="dt">NodeID =</span> <span class="st">&quot;name&quot;</span>,
               <span class="dt">width =</span> <span class="dv">700</span>, <span class="dt">fontSize =</span> <span class="dv">12</span>, <span class="dt">nodeWidth =</span> <span class="dv">30</span>)</code></pre></div>
<div id="htmlwidget-09d93928d80b1d542d2a" style="width:700px;height:480px;" class="sankeyNetwork html-widget"></div>
<script type="application/json" data-for="htmlwidget-09d93928d80b1d542d2a">{"x":{"links":{"source":[41,41,39,39,40,40,39,39,39,39,39,40,39,40,39,39,40,39,39,39,39,39,39,39,41,40,39,39,39,40,39,40,39,40,39,39,39,39,39,39,40,39,40,39,40,39,40,39,40,39,39,12,28,10,13,11,24,15,18,20,16,21,17,19,23,14,33,22,34,9,8,6,5,3,4,2,0,7,1,31,26,32,25,35,33,27,30,31,26,24,18,29,32,25,23,35,33,22,27,34,36,38,37],"target":[36,38,12,28,30,31,31,8,10,13,11,26,26,24,24,15,18,18,20,16,6,5,3,4,37,29,21,17,19,32,32,25,25,23,23,2,0,7,1,14,35,35,33,33,22,22,27,27,34,34,9,43,43,43,43,43,43,43,43,43,43,43,43,43,43,43,43,43,43,43,42,42,42,42,42,42,42,42,42,45,45,45,45,45,45,45,44,44,44,44,44,44,44,44,44,44,44,44,44,44,46,46,46],"value":[9,8,10,10,6,8,2,9,10,10,10,3,7,2,8,10,2,8,10,10,10,10,7,10,19,10,7,9,5,8,2,10,1,1,9,9,10,10,9,10,3,1,4,4,8,3,6,2,4,1,9,10,10,10,10,10,8,10,8,10,10,7,9,5,9,10,1,3,1,9,9,10,10,7,10,9,10,10,9,2,7,2,1,1,3,2,6,8,3,2,2,10,8,10,1,3,4,8,6,4,9,8,19]},"nodes":{"name":["88","9","84","175","177","173","171","89","159","SFr","160","162","12","161","93","165","169","58","166","64","168","51","Const","77","164","75","163","ESan","153","48","156","157","73","Aqu","Mat","98","101","32","102","Peninsula","Cape","Mainland","NBP-C","CBP-C","SCBP-A","SBP-C","SON-B"],"group":["88","9","84","175","177","173","171","89","159","SFr","160","162","12","161","93","165","169","58","166","64","168","51","Const","77","164","75","163","ESan","153","48","156","157","73","Aqu","Mat","98","101","32","102","Peninsula","Cape","Mainland","NBP-C","CBP-C","SCBP-A","SBP-C","SON-B"]},"options":{"NodeID":"name","NodeGroup":"name","LinkGroup":null,"colourScale":"d3.scaleOrdinal(d3.schemeCategory20);","fontSize":12,"fontFamily":null,"nodeWidth":30,"nodePadding":10,"units":"","margin":{"top":null,"right":null,"bottom":null,"left":null},"iterations":32,"sinksRight":true}},"evals":[],"jsHooks":[]}</script>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="parentage-analyses.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="population-models.html" class="navigation navigation-next " aria-label="Next page""><i class="fa fa-angle-right"></i></a>

<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": false,
"twitter": false,
"google": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "weibo", "instapaper"],
"instapaper": false
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "section",
"scroll_highlight": true
},
"search": true
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    if (location.protocol !== "file:" && /^https?:/.test(script.src))
      script.src  = script.src.replace(/^https?:/, '');
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
